%! BibTeX Compiler = biber
%TC:ignore
\documentclass{article}
\usepackage{caption}
\usepackage{censor}
\usepackage{xcolor, colortbl}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\definecolor{LIGHTGREY}{gray}{0.9}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
    linkcolor=DARKBLUELINK,
    anchorcolor=DARKBLUELINK,
    citecolor=DARKBLUELINK,
    filecolor=DARKBLUELINK,
    menucolor=DARKBLUELINK,
    urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage{biorxiv}
\usepackage[backend=biber,eprint=false,isbn=false,url=false,giveninits=true, style=apa,date=year]{biblatex}
\DeclareDelimFormat{nameyeardelim}{\addcomma\space}
\addbibresource{references.bib}

\usepackage{url}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{lmodern}
\usepackage{xfrac, nicefrac}
\usepackage{bm}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{graphicx}
\usepackage{bbold}
\usepackage{pdfpages}
\pdfinclusioncopyfonts=1
\usepackage{lineno}
\usepackage{tabu}
\usepackage{hhline}
\usepackage{multicol,multirow,array}
\usepackage{etoolbox}
\usepackage{booktabs}
\usepackage{makecell}
\usepackage{marvosym}

% -- Defining colors:
\definecolor{backcolour}{rgb}{0.95,0.95,0.92}% Definig a custom style:
\lstdefinestyle{mystyle}{
    backgroundcolor=\color{backcolour},
    basicstyle=\ttfamily\scriptsize\bfseries,
    breakatwhitespace=false,
    breaklines=true,
    captionpos=t,
    keepspaces=true,
    showspaces=false,
    showstringspaces=false,
    showtabs=false,
    tabsize=2
}% -- Setting up the custom style:
\lstset{style=mystyle}
\captionsetup[table]{hypcap=false}
\captionsetup[figure]{hypcap=false}

\newcommand{\defEqual}{\stackrel{\text{def}}{=}}
\newcommand{\Multiply}{\cdot}
\newcommand{\MultiplyMatrix}{\times}
\newcommand{\UniDimArray}[1]{\bm{#1}}
\newcommand{\BiDimArray}[1]{\bm{#1}}
\newcommand{\tr}{^{\intercal}}
\newcommand{\inv}{^{-1}}
\DeclareMathOperator{\E}{\mathbb{E}}
\DeclareMathOperator{\Var}{\text{Var}}
\DeclareMathOperator{\Std}{\text{Std}}
\DeclareMathOperator{\Cov}{\text{Cov}}
\newcommand{\Qst}{Q$_\text{ST}$}
\newcommand{\Fst}{F$_\text{ST}$}
\newcommand{\QstFst}{\Qst--\Fst}
\newcommand{\der}{\mathrm{d}}
\newcommand{\e}{\text{e}}
\newcommand{\Ne}{N_{\text{e}}}
\newcommand{\dnds}{\omega}
\newcommand{\pn}{\pi_N}
\newcommand{\ps}{\pi_S}
\newcommand{\pnps}{\pn / \ps}
\newcommand{\proba}{\mathbb{P}}
\newcommand{\pfix}{\proba_{\text{fix}}}
\newcommand{\Spi}{i}
\newcommand{\Spj}{j}
\newcommand{\NbrTaxa}{n}
\newcommand{\Time}{t}
\newcommand{\Trait}{P}
\newcommand{\Heritability}{h^2}
\newcommand{\HeritabilitySpi}{\Heritability_{\Spi}}
\newcommand{\MeanTrait}{\bar{\Trait_{\Time}}}
\newcommand{\VecTrait}{\UniDimArray{\bar{\Trait}}}
\newcommand{\Root}{0}
\newcommand{\RootTrait}{\Trait_{\Root}}
\newcommand{\VarPhy}{\Var \left[\MeanTrait\right]}
\newcommand{\StdPhy}{\Std \left[\MeanTrait\right]}
\newcommand{\VecZero}{\UniDimArray{0}}
\newcommand{\VecOne}{\UniDimArray{1}}
\newcommand{\Distance}{\BiDimArray{D}}
\newcommand{\DistanceMatrix}{\BiDimArray{\Distance}}
\newcommand{\MutationRatePheno}{\mu}
\newcommand{\MutationRateNuc}{u}
\newcommand{\SubRate}{q}
\newcommand{\NbrLoci}{L}
\newcommand{\VarPhenotype}{V_{\Trait}}
\newcommand{\VarPhenotypeSpi}{V_{\Trait, \Spi}}
\newcommand{\VarGenetic}{V_{\mathrm{A}}}
\newcommand{\VarGeneticSpi}{V_{\mathrm{A}, \Spi}}
\newcommand{\VarEnv}{V_{\mathrm{E}}}
\newcommand{\VarMutation}{V_{\mathrm{M}}}
\newcommand{\GenArchi}{\NbrLoci \Multiply \E \left[ a^2 \right]}
\newcommand{\RateBetween}{\sigma^2_{\mathrm{B}}}
\newcommand{\RateWhithin}{\sigma^2_{\mathrm{W}}}
\newcommand{\RateWhithinSpi}{\sigma^2_{\mathrm{W}, \Spi}}
\newcommand{\VecRateWhithin}{\UniDimArray{\RateWhithin}}
\newcommand{\EstRateBetween}{\widehat{\RateBetween}}
\newcommand{\EstRateWhithin}{\widehat{\RateWhithin}}
\newcommand{\NI}{\rho}
\newcommand{\EstNI}{\widehat{\rho}}
\newcommand{\StdSelection}{\sigma}
\newcommand{\VarSelection}{\StdSelection^2}

% Tree
\newcommand{\Nbranch}{2 \NbrTaxa - 2}
\newcommand{\WishartPostDf}{2 \NbrTaxa + 1}
\newcommand{\Ntrait}{K}
\newcommand{\contrast}{\UniDimArray{C}}
\newcommand{\Covariancematrix}{\Sigma}
\newcommand{\CovarianceMatrix}{\BiDimArray{\Covariancematrix}}
\newcommand{\Precisionmatrix}{\Omega}
\newcommand{\PrecisionMatrix}{\BiDimArray{\Precisionmatrix}}
\newcommand{\Identitymatrix}{\BiDimArray{I}}
\newcommand{\brownian}{\mathcal{B}}
\newcommand{\Brownian}{\UniDimArray{\brownian}}
\newcommand{\Scattermatrix}{\BiDimArray{A}}
\newcommand{\Multivariate}{\UniDimArray{Z}}

\renewcommand{\baselinestretch}{1.5}
\renewcommand{\arraystretch}{1.2}
\linenumbers
\frenchspacing

\title{Detecting diversifying selection for a trait from within and between-species genotypes and phenotypes}
\rhead{\scshape Trait selection from within and between-species variation}

\author{~}
\begin{document}

\maketitle

% Abstract (â‰¤ 250 words)
%TC:endignore
\begin{abstract}
    To quantify selection acting on a trait, methods have been developed using either within or between-species variation.
    However, methods using within-species variation do not integrate the changes at the macro-evolutionary scale.
    Conversely, current methods using between-species variation usually discard within-species variation, thus not accounting for processes at the micro-evolutionary scale.
    The main goal of this study is to define a neutrality index for a quantitative trait, by combining within- and between-species variation.
    This neutrality index integrates nucleotide polymorphism and divergence for normalizing trait variation.
    As such, it does not require estimation of population size nor of time of speciation for normalization.
    Our index can be used to seek deviation from the null model of neutral evolution, and test for diversifying selection.
    Applied to brain mass and body mass at the mammalian scale, we show that brain mass is under diversifying selection.
    Finally, we show that our test is not sensitive to the assumption that population sizes, mutation rates and generation time are constant across the phylogeny, and automatically adjust for it.
\end{abstract}

\keywords{Quantitative genetics \and Trait evolution \and Selection \and Phylogenetics \and Population genetics }

% Research Article (7500 words)
\section{Introduction}\label{sec:introduction}

Determining whether a trait is under a particular regime of selection has been a long-standing goal in evolutionary biology.
Fundamentally, distinguishing neutral evolution from selection requires determining which selective regime is supported by the observed variation of traits or sequences.
The variation of phenotypes (traits) and genotypes (sequences) can be observed at different scales, across different development stages at the individual level, across different individuals and populations at the species level, and finally across different species at the phylogenetic level.
All these systems require different assumptions and methodologies, and the endeavor to determine the selective regime for a given trait has thus incorporated theories, methods, and developments across various fields of evolutionary biology such as quantitative genetics, population genetics, phylogenetics and comparative genomics~\parencite{lynch_genetics_1998, walsh_evolution_2018}.

Leveraging individual variations within the same species, Genome-Wide Association Studies (GWAS) in humans have shown that traits are mostly polygenic (many loci associated with a given trait) and under stabilizing selection, while the loci affecting those traits are mostly pleiotropic (many traits associated with a given locus) with additive effects~\parencite{simons_population_2018, sella_thinking_2019}.
Across several populations, by contrasting both trait and genetic differentiation, \QstFst\ methods have been used to determine the selective regime and to quantify the strength of selection acting on a trait~\parencite{merila_comparison_2001, leinonen_comparative_2008}.
A trait differentiation (\Qst) higher than genetic differentiation (\Fst) is interpreted as a signature of diversifying selection due to adaptation to different optimum trait values in the different populations~\parencite{lamy_qst_2012}.
Contrarily, \Qst\ lower than \Fst\ is interpreted as a signature of stabilizing selection.
However, it has been found that \QstFst\ methods require many populations~\parencite{ohara_bias_2005}, and that various factors can generate a spurious signal of selection~\parencite{pujol_are_2008, edelaar_comparisons_2011}.
Moreover, the test for diversifying selection is limited to recent local adaptation since the test is based on the variation observed within a single species.

% Pair of species
To disentangle selection from neutral evolution, trait variation can also be observed at a larger time scale.
For example, change in mean trait value accumulates linearly with time of divergence from a sister species, and also proportionally to the trait variance~\parencite{lande_genetic_1980, turelli_heritable_1984}. % #I would add some examples here that do not use expression data, and then make a clear transition to the use of expression data
Empirically, this effect can be observed for genes with larger within-species variation in gene expression level, which exhibits a faster accumulation of divergence in mean expression level~\parencite{khaitovich_neutral_2004}.
Altogether, both the trait variance and the evolution in mean value can be used to test for trait selection in a pair of species~\parencite{walsh_evolution_2018}.
Analogously, in the context of protein-coding DNA sequences, leveraging within species variation and divergence to a sister species is the crux of the \textcite{mcdonald_adaptative_1991} test. In such a test, inflation of divergence to the sister species is compared to polymorphism within species, while neutral makers (usually synonymous sites) are used to determine the neutral expectation and thus are used for normalization.

% Phylogenetics (mean trait evolution)
Alternatively, by accounting for the underlying relationships between several species, the selective regime for a quantitative trait can also be tested at the phylogenetic scale~\parencite{felsenstein_phylogenies_1985}.
Under neutral evolution, the change in mean trait value along a given branch of the tree is normally distributed, with a variance proportional to divergence time~\parencite{hansen_translating_1996}.
As a result, the mean trait value can be modeled as a Brownian process branching at every node of the tree~\parencite{hansen_translating_1996, harmon_phylogenetic_2018}.
Reconstructing the trait variation along the whole phylogeny as a Brownian process can thus constitute a null model of neutral trait evolution.
Deviations from the assumptions of the Brownian process are however well known.
When trait variation is constrained because of optimum mean trait values across or between species, the pattern of evolution can be modeled by the Ornstein-Uhlenbeck processes, which is often interpreted as a signature of stabilizing selection~\parencite{catalan_drift_2019}.
Alternatively, a trend in the Brownian process (the tendency of a trait to evolve in a certain direction without fixed optimum) is interpreted as a signature of directional selection at the phylogenetic scale~\parencite{silvestro_early_2019}.
However, studies have shown that such comparative approaches are subject to different biases~\parencite{harmon_phylogenetic_2018}.
First, a trait under stabilizing selection for which the optimal trait value is also changing as a Brownian process will not deviate from a Brownian process, and thus be wrongly classified as neutral~\parencite{hansen_translating_1996}.
In other words, the better fit of a Brownian process does not necessarily constitute proof of the neutral model.
Second, even for a trait evolving under a neutral regime, the Ornstein-Uhlenbeck process might sometimes be statistically preferred over a Brownian process due to sampling artifacts~\parencite{silvestro_measurement_2015, cooper_cautionary_2016, price_detecting_2022}.
Those limitations, altogether with the use of mean trait estimates leaving out the variance in traits between individuals, easily generate misclassification of selection from methods at the phylogenetic scale.

% At the frontier Phylogenetics/Population-genetics
At the frontier between micro and macroevolution, comparative methods at the phylogenetic scale have acknowledged the importance of modeling within-species variation together with changes in mean trait value to either describe measurement errors~\parencite{lynch_methods_1991, hansen_interpreting_2012}, incorporate values for individuals~\parencite{felsenstein_comparative_2008} or to scale the rate of change in mean trait value~\parencite{kostikova_bridging_2016, gaboriau_multiplatform_2020, gaboriau_exploring_2023}.
Between to within-species variation has also been used to infer diversifying selection by estimating the ratio of between to within variation of many traits and test for deviation from the average ratio across traits~\parencite{rohlfs_modeling_2014, rohlfs_phylogenetic_2015}.
Here, our goal was again to use both variances between and within species to determine the selective regime of a quantitative trait.
We build a novel framework that integrates trait variation at the phylogenetic and population scales together with estimates of nucleotide sequence variations at both scales.
It allowed us to define an expected ratio of normalized variance between and within species while setting the threshold of this ratio for neutral, stabilizing, and diversifying selection.
The ratio that we propose can be considered as a neutrality index for a any quantitative trait articulating trait and nucleotide variation within and between species.
Importantly, our neutrality index also leverages nucleotide divergence and polymorphism to normalize trait variation at both scales, such that it does not require estimating population size (within-species) or speciation time (between species).
From the field of population genetics, our study can be seen as the macro-evolutionary generalization of \QstFst\ methods to account for phylogenetic relationships between species.
From the field of phylogenetics, our study can be seen as an alternative to the EVE model~\parencite{rohlfs_modeling_2014, rohlfs_phylogenetic_2015} for a single trait, where we set a threshold for neutral evolution by leveraging species nucleotide polymorphism and divergence.
% Basically, we are testing whether the variance of means is greater than the mean of variances


\section{Materials and Methods}\label{sec:materials-and-methods}
\subsection{Neutrality index for a quantitative trait}\label{subsec:neutrality-index-for-a-quantitative-trait}

While observing trait variations across individuals of several species, we ask if the variation within species compared to variation between species is compatible with neutral evolution or not.
In statistical terms, this can also be framed as: Is the variance of means equal to the mean of variances?
The difficulty in such a study is that individuals are not independent samples, but are from species that diverged at different times, and each species has a putatively different population size.
By reviewing theoretical expectations and leveraging nucleotide sequence variations, the goal of this section is thus to obtain normalized trait variation between and within species that are equal if the trait is neutral.
Here we denote these normalized trait variations as respectively $\RateWhithin$ for within species and as $\RateBetween$ for between species.

\subsubsection{Within-species trait variations}

For a given trait, the genetic architecture is mainly defined by the number of loci encoding the trait ($\NbrLoci$) and the random additive effect of a mutation on the trait ($a$).
At the individual level, the mutational variance ($\VarMutation$) is the rate at which new mutations contribute to the trait variance per generation.
As shown in \textcite{lande_quantitative_1979, lande_sexual_1980}, $\VarMutation$ is a function of the mutation rate per loci per generation ($\MutationRatePheno$) and the genetic architecture of the trait as:
\begin{gather}
    \VarMutation = 2 \MutationRatePheno \Multiply \GenArchi \label{eq:var-mutation}.
\end{gather}

While in an infinitesimal model mutations supply new genetic variants, random genetic drift depletes standing variation~\parencite{turelli_commentary_2017, barton_infinitesimal_2017, sella_thinking_2019}.
For a neutral trait at equilibrium between mutation and drift~\parencite{lynch_mutation_1998}, the additive genetic variance in a species ($\VarGenetic$) is a function of the mutational variance ($\VarMutation$) and the effective number of individuals in the population ($\Ne$):
\begin{align}
    \VarGenetic & =  2 \Ne \Multiply \VarMutation, \\
    & = 4 \Ne \Multiply \MutationRatePheno \Multiply \GenArchi \text{ from eq.~\ref{eq:var-mutation}}\label{eq:var-genetic}.
\end{align}

For any neutral genomic region of interest, the nucleotide diversity, $\pi$, is measured as the number of mutations segregating in the population divided by the length of the region.
Any segregating mutations will eventually reach fixation or extinction due to random genetic drift and $\pi$ is also at a balance between mutations and drift.
As shown in \textcite{tajima_statistical_1989}, $\pi$ is a function of the mutation rate per loci per generation ($\MutationRateNuc$) and the effective population size ($\Ne$):
\begin{gather}
    \pi = 4 \Ne \Multiply \MutationRateNuc \label{eq:genetic-diversity}.
\end{gather}

We define $\RateWhithin$ as the ratio of additive genetic variance of the trait ($\VarGenetic$) over $\pi$ of any neutral genomic region of interest.
This ratio allows removing the effect of $\Ne$, giving $\RateWhithin$ as a proxy of the underlying genetic architecture:
\begin{align}
    \RateWhithin & \defEqual \frac{\VarGenetic }{\pi} \label{eq:def-rate-pop}, \\
    & = \frac{4 \Ne \Multiply \MutationRatePheno \Multiply \GenArchi}{4 \Ne \Multiply \MutationRateNuc} \text{ from eq.~\ref{eq:var-mutation} and~\ref{eq:genetic-diversity}}, \\
    & = \frac{\MutationRatePheno \Multiply \GenArchi}{\MutationRateNuc}. \label{eq:rate-pop}
\end{align}

The additive genetic variance is also equal to the observed phenotypic variance ($\VarPhenotype$) multiplied by narrow-sense heritability ($\Heritability$; \parencite{hill_data_2008}), as:
\begin{align}
    \VarGenetic & = \Heritability \Multiply \VarPhenotype.\label{eq:var-gen-pheno}
\end{align}
Which leads to $\RateWhithin$ being a function of $\VarPhenotype$ and $\Heritability$ as:
\begin{align}
    \RateWhithin & = \frac{\Heritability \Multiply \VarPhenotype }{\pi } \text{ from definition eq.~\ref{eq:def-rate-pop} and~\ref{eq:var-gen-pheno}}, \label{eq:def-rate-pheno-pop} \\
    & = \frac{\MutationRatePheno \Multiply \GenArchi}{\MutationRateNuc} \text{ from eq.~\ref{eq:rate-pop}} \label{eq:rate-pheno-pop}
\end{align}

\subsubsection{Between-species trait variations}

For a given species, we denote by $\MeanTrait$ the mean value of the trait across the individuals in the species at generation $\Time$.
If the trait is neutral and encoded by many loci as assumed by the infinitesimal model, $\MeanTrait$ evolves as a Brownian process~\parencite{felsenstein_phylogenies_1985, hansen_translating_1996}.
The variance of $\MeanTrait$ after $\Time$ generations, $\VarPhy$ is given \parencite{hansen_translating_1996} by:
\begin{align}
    \VarPhy & = \frac{\VarGenetic}{\Ne} \Multiply \Time \\
    & = 4 \Time \Multiply \MutationRatePheno \Multiply \GenArchi \label{eq:covar},\text{ from eq.~\ref{eq:var-genetic}},
\end{align}

Moreover, for any genomic region under neutral evolution, some mutations will eventually reach fixation due to random genetic drift, resulting in a substitution of a nucleotide at the species level.
The probability of fixation ($\pfix$) of a neutral mutation is $1/2\Ne$ \parencite{kimura_probability_1962}.
We can derive the substitution rate per generation $\SubRate$ as the number of mutations per generation ($2\Ne \Multiply \MutationRateNuc$) multiplied by the probability of fixation for each newly arisen mutations $\pfix$~\parencite{mccandlish_modeling_2014}, giving:
\begin{align}
    \SubRate & = 2 \Ne \Multiply \MutationRateNuc \Multiply \pfix, \\
    & = 2 \Ne  \Multiply \MutationRateNuc  \Multiply \dfrac{1}{2\Ne}, \\
    & = \MutationRateNuc. \label{eq:substitution-rate}
\end{align}
That is, if mutations are neutral, the rate of substitution within a genomic region equals the rate at which new mutations arise per generation for the same genomic region~\parencite{kimura_evolutionary_1968}.

After $\Time$ generations and assuming that no multiple substitutions occurred at the same site, the nucleotide divergence $d$, which is the fraction of the genomic region that generated a substitution, will be $\Time$ multiplied by the nucleotide substitution rate per generation ($\SubRate$):
\begin{align}
    d & = \Time \Multiply \SubRate \\
    & = \Time \Multiply \MutationRateNuc \text{ from eq.~\ref{eq:substitution-rate}}. \label{eq:distance}
\end{align}
We define $\RateBetween$ as the variance in the mean trait value ($\VarPhy$) normalized by 4 times the nucleotide divergence of any neutral genomic region ($4d$).
This ratio allows removing the effect of $\Time$, giving $\RateBetween$ as another proxy of the underlying genetic architecture:
\begin{align}
    \RateBetween & \defEqual \frac{\VarPhy}{4 d} \label{eq:def-rate-phy},\\
    & = \frac{4 \Time \Multiply \MutationRatePheno \Multiply \GenArchi}{4 \Time \Multiply \MutationRateNuc}\text{ from eq.~\ref{eq:covar} and~\ref{eq:distance}}, \\
    & = \frac{\MutationRatePheno \Multiply \GenArchi}{\MutationRateNuc} \label{eq:rate-phy}.
\end{align}

Altogether, starting from the same ancestral population, divergent lineages accumulate both phenotypic and nucleotide changes reaching fixation in the population.
These changes ultimately result in differences across lineages.
In equation \ref{eq:rate-phy}, we show that the variance in mean trait value between lineages ($\VarPhy$) does increase linearly with nucleotide divergence ($d$), if the trait and sequences are neutrally evolving and the genetic architecture of the trait has not changed.
In other words, the standard deviation in mean phenotypes across lineages ($\StdPhy$) becomes proportional to the square root of nucleotide divergence since the ancestral population ($\sqrt{d}$).

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=\textwidth, page=1] {figure1}
    \caption{
        Between species, the change along the phylogeny of the mean phenotypic trait allows the estimation of between-species trait variation, $\EstRateBetween$, which is normalized by nucleotide divergence.
        Within species, the genetic variance allows the estimation of within-species trait variation, $\EstRateWhithin$, which is  normalized by nucleotide diversity.
        $\EstNI$ is the ratio of $\EstRateBetween$ over $\EstRateWhithin$.
        Under neutral evolution, $\EstNI$ is expected to be equal to one.
        Under diversifying selection, the trait is heterogeneous between species, but homogeneous within species, leading to $\EstNI$ greater than one.
        Under stabilizing selection, the trait is homogeneous between species, leading to $\EstNI$ smaller than one.
        Importantly, the sequence from which nucleotide diversity and divergence are estimated should be neutrally evolving, but they are not necessarily linked to the quantitative trait under study, they allow for discarding the confounding effect on mutation rate diversity, population size and divergence time.
    }
    \label{fig:methods}
\end{figure*}

\subsubsection{Neutrality index}

The variability between either individuals or species can be obtained for both quantitative traits and genomic sequences.
At the population level, the variability of the trait between individuals can be combined with the nucleotide diversity of any neutrally evolving genomic region to obtain $\RateWhithin$.
At the phylogenetic level, the variability of the mean trait value between species can be combined with the nucleotide divergence of any neutrally evolving genomic region to obtain $\RateBetween$.
If the trait is neutrally evolving and the genetic architecture of the trait has not changed along the phylogenetic tree, we thus have:
\begin{align}
    \frac{\RateBetween}{\RateWhithin} &= \frac{\VarPhy}{4 d} \Multiply \frac{\pi}{\Heritability \Multiply \VarPhenotype} \text{ by definition and eq.~\ref{eq:def-rate-pheno-pop} and~\ref{eq:def-rate-phy}}\\,
 &= \frac{\MutationRatePheno \Multiply \GenArchi}{\MutationRateNuc} \Multiply \frac{\MutationRateNuc}{\MutationRatePheno \Multiply \GenArchi} \text{ from eq.~\ref{eq:rate-pheno-pop} and~\ref{eq:rate-phy}}, \\
    &= 1.
\end{align}
We define a neutrality index $\NI$ as:
\begin{align}
\NI \defEqual \frac{\RateBetween}{\RateWhithin},
\end{align}
 which will equal to $1$ for a trait evolving neutrally.
Both $\RateBetween$ and $\RateWhithin$ can be estimated using quantitative trait and genomic sequences within and between species, while neither the mutation rates ($\MutationRatePheno$ and $\MutationRateNuc$), nor the effective population size ($\Ne$) or time of divergence ($\Time$) need to be estimated.
Moreover, the sequence from which $\pi$ and $d$ are estimated should be neutrally evolving, but they are not necessarily linked to the quantitative trait under study.

\subsection{Estimation}\label{subsec:estimate}

We denote $\EstNI$, $\EstRateBetween$ and $\EstRateWhithin$ the point estimates of respectively $\NI$, $\RateBetween$ and $\RateWhithin$.
At the phylogenetic scale, $\EstRateBetween$ (defined eq. \ref{eq:def-rate-phy}) can be seen as an estimate of the rate of the evolution of the quantitative trait along a phylogeny, when the tree is measured in unit of $4d$ instead of time as is usually the case in the comparative framework~\parencite{felsenstein_phylogenies_1985, omeara_testing_2006}.
We provide a maximum likelihood estimate for $\NI$ as well as a Bayesian estimate to derive posterior probabilities that the null model of neutrality (i.e.~$\NI = 1$) is rejected.

\subsubsection{Maximum likelihood estimate}
At the phylogenetic scale, for $\NbrTaxa$ taxa in the tree, $\DistanceMatrix$ ($\NbrTaxa \times \NbrTaxa$) is the distance matrix computed from the branch lengths ($d$ as nucleotide divergence in units of substitutions per site) and the topology of the phylogenetic tree.
The diagonal $\Distance_{\Spi,\Spi}$ represents the total distances from the root of the tree to each taxon ($\Spi$).
The off-diagonal elements ($\Distance_{\Spi,\Spj} = \Distance_{\Spj,\Spi}$) are the distances between the root and the most recent common ancestor of taxa $\Spi$ and $\Spj$.
The state $\RootTrait$ at the root of the tree for the trait can be estimated from the $\NbrTaxa \times 1$ vector of mean trait values $\VecTrait$ at the tips of the tree using maximum likelihood~\parencite{omeara_testing_2006}:
\begin{gather}
    \RootTrait = \left( \VecOne\tr \MultiplyMatrix \DistanceMatrix\inv \MultiplyMatrix \VecOne \right)\inv \Multiply \left( \VecOne\tr \MultiplyMatrix \DistanceMatrix\inv \MultiplyMatrix \VecTrait \right), \label{eq:estimated-root-trait}
\end{gather}
where $\VecOne$ is an $\NbrTaxa \times 1$ column vector of ones.

Finally, between-species variation $\EstRateBetween$ is estimated as~\parencite{omeara_testing_2006}:
\begin{gather}
    \EstRateBetween = \frac{1}{4}\frac{\left( \VecTrait -  \RootTrait \Multiply \VecOne \right)\tr \MultiplyMatrix \DistanceMatrix\inv \MultiplyMatrix \left( \VecTrait -  \RootTrait \Multiply \VecOne  \right)}{\NbrTaxa - 1}. \label{eq:estimated-rate-phy}
\end{gather}

For a given species $\Spi$ with inter-individual data available, additive genetic variance of a trait ($\VarGeneticSpi$) is the product of heritability ($\Heritability_{i}$) and phenotypic variance ($V_{\Trait, i}$).
The ratio of $\VarGeneticSpi$ over nucleotide diversity of neutrally evolving sequences ($\pi_{\Spi}$) is a sample estimate of $\RateWhithin$.
Averaged across all species, we obtain the estimate $\EstRateWhithin$ as:
\begin{gather}
    \EstRateWhithin = \dfrac{1}{\NbrTaxa}\sum_{i=1}^{\NbrTaxa}\frac{  \VarGeneticSpi}{ \pi_{i}} = \dfrac{1}{\NbrTaxa}\sum_{i=1}^{\NbrTaxa} \frac{  V_{\Trait, i} \Multiply \Heritability_{i}}{ \pi_{i}}. \label{eq:estimated-rate-pop}
\end{gather}

As depicted in fig.~\ref{fig:methods}, the neutrality index is estimated as:
\begin{gather}
    \EstNI = \frac{\EstRateBetween}{\EstRateWhithin}. \label{eq:estimated-NI}
\end{gather}

\subsubsection{Multivariate Brownian process}

In the previous section, $\EstNI$ is estimated independently for each trait of interest.
Here we generalize to $\Ntrait$ traits co-varying along the phylogenetic tree.
Trait variation along the phylogenetic tree is modeled as a $\Ntrait$-dimensional Brownian process $\Brownian$ ($1 \times \Ntrait$) starting at the root and branching along the tree topology~\parencite{huelsenbeck_detecting_2003, lartillot_phylogenetic_2011, lartillot_joint_2012, latrille_inferring_2021}.
The rate of change of the Brownian process is determined by the positive semi-definite and symmetric covariance matrix between traits $\CovarianceMatrix$ ($\Ntrait \times \Ntrait$).
Importantly, the branch length of the tree on which the Brownian process runs is measured in units of $4d$ ($d$ is the nucleotide divergence), and not time as is usually the case in the comparative method. The off-diagonal elements of $\CovarianceMatrix$ are the covariance between traits, and the diagonal elements are the variance of each trait when measured in $4d$ units, and thus equate to $\EstRateBetween$ (see section~S\ref{subsec:multivariate-brownian-process}).


\subsubsection{Bayesian estimate}
The Bayesian framework allows obtaining the posterior distribution of neutrality index ($\EstNI$) for traits of interest.
We used the \textit{BayesCode} software to model $\Ntrait$-dimensional Brownian processes along a phylogenetic tree~\parencite{latrille_inferring_2021}.
With an inverse Wishart distribution as the {prior} on the covariance matrix, the {posterior} on $\CovarianceMatrix$, conditional on $\brownian$ is also an invert Wishart distribution (see section~S\ref{subsec:sampling-the-covariance-matrix}).
We used Metropolis-Hastings algorithm to sample $\Brownian$, while the posterior distribution of $\CovarianceMatrix$ is sampled using Gibbs sampling.
For each trait and each species, the prior on heritability ($\Heritability$) for each species is set as a uniform distribution with user-defined boundaries.
Heritability and phenotypic variance for each trait are combined with nucleotide diversity to compute $\EstRateWhithin$ for each species before being averaged across species (as in eq.~\ref{eq:estimated-rate-pop}).
From $\EstRateWhithin$ and $\CovarianceMatrix$, the posterior distribution of $\EstNI$ (as in eq.~\ref{eq:estimated-NI}) is obtained for each trait.
The posterior distribution of $\EstNI$ thus allows testing for deviation from neutrality (Fig.~\ref{fig:methods}), for example, by computing $\proba [\EstNI > 1 ]$ to test for evidence of diversifying selection and $\proba [\EstNI < 1 ]$ to test for evidence of stabilizing selection.

\subsubsection{Applicability to empirical data}

Our method assumes that the narrow-sense heritability ($\Heritability$) of a trait is known such as to estimate additive genetic variance ($\VarGenetic$) from phenotypic variance ($\VarPhenotype$) as $\VarGenetic = \Heritability \Multiply \VarPhenotype$.
Fortunately, if heritability is not known, the test for diversifying selection can still be performed, although it is underpowered.
Indeed, if the additive genetic variance is substituted by phenotypic variance, it is equivalent to assuming complete heritability ($\Heritability = 1$).
Because $\Heritability \leq 1$ by definition, we overestimate the within-species variation and thus underestimate $\EstNI$.
It is, however, possible to test for diversifying selection because testing for $\EstNI > 1$ while using phenotypic variance instead of additive genetic variance means that knowing the additive genetic variance would have only increased the evidence for diversifying selection.
Similarly, using the broad-sense heritability ($H^2$) instead of narrow-sense heritability ($\Heritability$) results in an underestimation of $\EstNI$ since $\Heritability \leq H^2$.
In contrast, the test for stabilizing selection is invalid if $\EstNI$ is underestimated.
Several assumptions made by our test might not hold on empirical data and their consequences on the neutrality index and the test that can be performed are shown in Table~\ref{table:assumptions}.

\subsection{Simulation}\label{subsec:simulations}

% vG: 13.8, vP: 69.0, vE: 55.2, nbr_loci: 5000, a: 1.0, mut_rate: 1.38e-05, pop_size: 50, h2: 0.2
% pS = 0.0027577478571428568
% Tree length (dS) = 2.340579
% Tree length (My) = 1259.4490200000002
% Root age (My) = 98.9489413888889
% Root age (dS) = 0.18388820079995308
% nbr_sites_var = 27.577478571428568
% u = 1.3788739285714283e-05
% nbr_generations = 13336.114128321291
We tested the performance of our neutrality index ($\NI$) to detect selection on a quantitative trait using simulations.
We performed simulations under different selective regimes (neutral, stabilizing, diversifying), different demographic histories (constant or fluctuating population size) and different evolution of the mutation rate (constant or fluctuating).
Simulations were individual-based and followed a Wright-Fisher model with mutation, selection and drift for a diploid population including speciation along a predefined ultrametric phylogenetic tree (Fig.~\ref{fig:simulator}A\&B).
Each individual phenotypic value was the sum of genotypic value and an environmental effect.
The environmental effect was normally distributed with variance $\VarEnv$.
We assumed that the genotypic value was encoded by $\NbrLoci=5,000$ loci, with each locus contributing an additive effect that was normally distributed with standard deviation $a=1$ (Fig.~\ref{fig:simulator}A and section~S\ref{subsec:genotype-phenotype-map} for theoretical formulation).
We assumed a trait with a narrow-sense heritability of $\Heritability=0.2$ and computed the theoretical $\VarEnv$ accordingly (see section~S\ref{subsec:genotype-phenotype-map}).
Assuming a diploid panmictic population of size $\Ne=50$ at the root of the tree, and with non-overlapping generations, we simulated explicitly each generation along an ultrametric phylogenetic tree.
For each offspring, the number of mutations was drawn from a Poisson distribution with mean $2 \Multiply \MutationRatePheno \Multiply \NbrLoci $, with the mutation rate per locus per generation $\MutationRatePheno$.
From the empirical mammalian dataset (see next section), we computed an average nucleotide divergence from the root to leaves of $0.18$ and average genetic diversity of $0.00276$.
We scaled parameters in our simulations to fit plausible values for mammals.
We thus used a nucleotide mutation rate of $\MutationRateNuc=0.00276 / 4 \Ne = 1.38 \times 10^{-5}$ per generation per locus and a total of $\Time = 0.18 / 1.38 \times 10^{-5} = 13,500$ generations from root to leaves, and the number of generations along each branch was proportional to the branch length.
We set $\MutationRatePheno=\MutationRateNuc$ without loss in generality since the genetic architecture ($\NbrLoci$ and $a$) is assumed constant in the simulator.

The changes in log-$\MutationRatePheno$ and log-$\Ne$ along the lineages were both modeled by a geometric Brownian process ($\brownian \left(0, \sigma_{\MutationRatePheno}=0.0086\right)$ and $\brownian \left(0, \sigma_{\Ne}=0.0086\right)$, which led to a standard deviation of $0.0086 \Multiply \sqrt {13,500} = 1.0$ in log-space from root to leaves.
An Ornstein-Uhlenbeck process was overlaid to the instant value of log-$\Ne$ provided by the geometric Brownian process to account for short-term changes between generations ($\text{OU} \left(0, \sigma_{\Ne}=0.1, \theta_{\Ne}=0.9\right)$).
The geometric Brownian motion accounted for long-term fluctuations (low rate of changes $\sigma_{\Ne}$ but unbounded), while the Ornstein-Uhlenbeck introduced short-term fluctuations (high rate of changes $\sigma_{\Ne}$ but bounded and mean-reverting).
The simulation started from an initial sequence at equilibrium at the root of the tree and, at each node, the process was split until it finally reached the leaves of the tree.
From a speciation process perspective, this was equivalent to an allopatric speciation over one generation.

At each generation, parents were randomly sampled with a weight proportional to their fitness ($W$).
Selection was modeled as a one-dimensional Fisher's geometric landscape, with the fitness of an individual being a monotonously decreasing function of the distance between the individual and the optimal phenotype~\parencite{tenaillon_utility_2014,blanquart_epistasis_2016}.
More specifically, the fitness of an individual was given by $W = \e^{(\Trait - \lambda)^2/ \alpha}$, where $\Trait$ was the trait value of the individual, $\lambda=0.0$ was the optimal trait value, and $\alpha=0.02$ was the strength of selection.
Mutations were considered as a displacement of the phenotype in the multidimensional space.
Beneficial mutations moved the phenotype closer to the optimum, while deleterious mutations moved it further away.
Stabilizing selection was implemented by fixing the optimum phenotype to a single value ($\lambda=0.0$).
Diversifying selection was implemented by allowing the optimum phenotype to move along the phylogenetic tree as a geometric Brownian process~\parencite{hansen_stabilizing_1997} ($\lambda \sim \brownian \left(0, \sigma_{\lambda}=1.0\right)$).
Neutral evolution was implemented by fixing the fitness landscape ($W=1$), which meant that each individual had the same probability of being sampled at each generation.

Nucleotide diversity ($\pi$) was measured as the heterozygosity of neutral markers that were simulated along the phylogenetic tree but not linked to the trait simulated.
Nucleotide divergence ($d$) was measured as the number of substitutions per site of neutral markers along the branches of the phylogenetic tree.
The additive genetic variance was measured as phenotypic variance multiplied by heritability.
Heritability was estimated from the slopes of the regression of offspring's phenotypic trait values on parental phenotypic trait values~\parencite{lynch_genetics_1998} averaged over the last 10 simulated generations.
Heritability was thus not a given parameter of the simulations, but rather measured as it would be in empirical data.

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=\textwidth, page=1] {figure2}
    \caption{
        Wright-Fisher simulations with mutation, selection and drift.
        Left panel: For a given individual, the trait phenotypic value is the sum of genotypic value and a environmental effect (standard deviation $\VarEnv$).
        The trait's genotypic value is encoded by $\NbrLoci$ loci, with each locus contributing additively to the genotypic value.
        Parents are selected for reproduction to the next generation according to their phenotypic value, with a probability proportional to their fitness.
        Mutations are drawn from a Poisson distribution, with each locus having a probability $\MutationRatePheno$ to mutate.
        Drift is modeled by the resampling of parents.
        Right panel: examples of a trait evolving along a phylogenetic tree, with the mean phenotype (black line) and the variance of the trait genotypic value (gray area).
    }
    \label{fig:simulator}
\end{figure*}

\subsection{Empirical dataset}\label{subsec:empirical-dataset}

We analyzed a dataset of body and brain masses from mammals.
The log-transformed values of body and brain masses were taken from \textcite{tsuboi_breakdown_2018}.
We removed individuals not marked as adults and split the data into males and females due to sexual dimorphism in body and brain masses.
We discarded species with only one representative sample.
The mammalian nucleotide diversity was obtained from the Zoonomia project~\parencite{genereux_comparative_2020}, with nucleotide divergence obtained on a set of neutral markers in \textcite{foley_genomic_2023}, and with nucleotide diversity measured as heterozygosity in \textcite{wilder_contribution_2023}.

We also analyzed a dataset of primate species, with the nucleotide variation obtained from \textcite{kuderna_global_2023} and the quantitative trait variation also from \textcite{tsuboi_breakdown_2018}, using the same filtering as for the mammalian dataset.
However, the primate nucleotide divergence was not obtained on a set of neutral markers as for the mammalian dataset, but across the whole genome.

\section{Results}\label{sec:results}

\subsection{Neutrality index}

For a neutral trait, the genetic architecture, meaning the number of loci encoding the trait and the average effect of a mutation on the trait, is formally related to both within and between-species variation of the trait.
We defined the neutrality index as $\NI = \RateBetween/\RateWhithin$, which equals $1$ for a neutral trait (see Materials and Methods), suggesting that traits for which this relationship was not verified were putatively under selection.
Under stabilizing selection, the variation between species is depleted because the mean trait value is maintained toward similar values between different species, which leads to $\NI < 1$.
In contrast, under diversifying selection, the variation between species is inflated because species will have potentially different trait values~\parencite{hansen_stabilizing_1997}, which leads to $\NI > 1$.
Our neutrality index for a quantitative trait leveraged the data for any number of species, and took advantage of the signal over the whole phylogenetic tree, while at the same time taking into account phylogenetic inertia and addressing the non-independence between species (Fig.~\ref{fig:methods}).
This statistic was obtained as a maximum likelihood estimate ($\EstNI$), from eq.~\ref{eq:estimated-rate-pop} and~\ref{eq:estimated-rate-phy}.
We also devised a Bayesian estimate to obtain the posterior distribution of the neutrality index, and test for diversifying selection as $\proba [\EstNI > 1]$, and stabilizing selection as $\proba [\EstNI < 1]$.

Our neutrality index made a series of assumptions that we described in details in the Material and Methods section.
Table \ref{table:assumptions} summarized these assumptions and outlined possible consequences for the neutrality test that we proposed.

\subsection{Results against simulations}\label{subsec:results-against-simulations}

The inference framework was first tested on independently simulated datasets matching an empirically relevant mammalian empirical regime (see Materials and Methods).
Under constant population size ($\Ne$) and constant mutation rates ($\MutationRatePheno$ and $\MutationRateNuc$) across the phylogenetic tree (fig.~\ref{fig:results-simulations}, top row), we found no false negative for simulations of stabilizing ($\proba [\EstNI < 1] > 0.975$; blue in fig.~\ref{fig:results-simulations}) or diversifying ($\proba [\EstNI > 1] > 0.975$; red in fig.~\ref{fig:results-simulations}) selection.
For simulations under neutral evolution, 77\% of those were correctly identified ($0.025 \leq \proba [\EstNI > 1] \leq 0.975$; yellow in fig.~\ref{fig:results-simulations}), while 21\% and 2\% were wrongly detected as stabilizing or diversifying selection, respectively.
Once we introduced fluctuating $\Ne$, $\MutationRatePheno$ and $\MutationRateNuc$(Fig.~\ref{fig:results-simulations}, bottom row), our ability to identify simulations under either diversifying or stabilizing selection remained the same with all cases detected correctly.
For simulations under neutral evolution, 51\% of the simulations were correctly detected ($0.025 \leq \proba [\EstNI > 1] \leq 0.975$), while 49\% were detected as stabilizing selection ($\proba [\EstNI < 1] > 0.975$) and none as diversifying selection.

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=\textwidth, page=1] {figure3}
    \caption{
        $10,000$ simulations of trait evolution along a phylogenetic tree under different selection regimes.
        Traits simulated under stabilizing selection (blue), under a neutral evolution (yellow), and under a moving optimum (red).
        Histogram of ratio of between-species trait variation ($\EstRateBetween$) over within-species trait variation $\EstRateWhithin$ with $\EstNI = \EstRateBetween / \EstRateWhithin$ estimated from each simulated data (left) and probabilities of $\EstNI$ being greater than $1$ (right).
        Effective population size ($\Ne$) and mutation rates ($\MutationRatePheno$ and $\MutationRateNuc$) were either constant (top row), or fluctuating as a Brownian process along the phylogenetic tree (bottom row).
    }
    \label{fig:results-simulations}
\end{figure*}


\begin{table*}[t!]
    \centering
    \begin{adjustbox}{width = 0.75\textwidth}
        \begin{tabular}{||l|l|l|c|c|c|c|c||}
        \toprule
        Dataset & Trait & $\Heritability$ & Sex & $\NbrTaxa$ & $\EstNI$ & $95\%$ CI for $\EstNI$ & $\proba [\EstNI > 1 ]$ \\ \hline
        \midrule
        Mammals & Body mass & 1.0 & \Male & 36 & 0.340 & 0.217-0.523 & 0.000 \\ \hline
        Mammals & Body mass & 1.0 & \Female & 26 & 0.277 & 0.160-0.490 & 0.000 \\ \hline
        Mammals & Body mass & $\mathcal{U}(0.2, 0.4)$ & \Male & 36 & 1.124 & 0.721-1.754 & 0.635 \\ \hline
        Mammals & Body mass & $\mathcal{U}(0.2, 0.4)$ & \Female & 26 & 0.936 & 0.523-1.715 & 0.324 \\ \hline \hline
        Mammals & Brain mass & 1.0 & \Male & 36 & 1.351 & 0.851-2.173 & 0.877 \\ \hline
        Mammals & Brain mass & 1.0 & \Female & 26 & 1.727 & 0.991-2.938 & 0.972 \\ \hline
        Mammals & Brain mass & $\mathcal{U}(0.2, 0.4)$ & \Male & 36 & 4.527 & 2.831-7.091 & 1.000 \\ \hline
        Mammals & Brain mass & $\mathcal{U}(0.2, 0.4)$ & \Female & 26 & 6.001 & 3.288-10.941 & 1.000 \\ \hline \hline
        Primates & Body mass & 1.0 & \Male & 71 & 0.558 & 0.401-0.784 & 0.000 \\ \hline
        Primates & Body mass & 1.0 & \Female & 65 & 0.389 & 0.278-0.547 & 0.000 \\ \hline
        Primates & Body mass & $\mathcal{U}(0.2, 0.4)$ & \Male & 71 & 1.875 & 1.288-2.695 & 1.000 \\ \hline
        Primates & Body mass & $\mathcal{U}(0.2, 0.4)$ & \Female & 65 & 1.296 & 0.899-1.821 & 0.914 \\ \hline \hline
        Primates & Brain mass & 1.0 & \Male & 71 & 1.929 & 1.395-2.616 & 1.000 \\ \hline
        Primates & Brain mass & 1.0 & \Female & 65 & 1.950 & 1.399-2.790 & 1.000 \\ \hline
        Primates & Brain mass & $\mathcal{U}(0.2, 0.4)$ & \Male & 71 & 6.479 & 4.658-8.944 & 1.000 \\ \hline
        Primates & Brain mass & $\mathcal{U}(0.2, 0.4)$ & \Female & 65 & 6.522 & 4.664-9.294 & 1.000 \\
        \bottomrule
        \end{tabular}
    \end{adjustbox}
    \caption{
    Test of diversifying selection on a mammal and a primate dataset, by splitting males (\Male) and females (\Female).
    Traits considered were body mass or brain mass (log-transformed).
    Heritability ($\Heritability$) was either assumed complete ($\Heritability=1.0$) or uniformly distributed between 20\% and 40\%  ($\Heritability \sim \mathcal{U}(0.2, 0.4)$).
    $\NbrTaxa$ was the number of species in the dataset.
    $\EstNI$ was the posterior estimate of our neutrality index, with the $95\%$ credible interval (CI) for $\EstNI$ also computed.
    $\proba [\EstNI > 1 ]$ was the estimated posterior probability of diversifying selection.
    }
    \label{table:empirical}
\end{table*}

\subsection{Results on empirical data}\label{subsec:results-on-empirical-data}

For mammalian body and brain mass, we obtained male (\Male) and female (\Female) trait variations.
Combined with nucleotide diversity and divergence, we estimated $\EstNI$ and posterior probabilities of diversifying selection under different assumptions for trait heritability as shown in the Table~\ref{table:empirical}.
Assuming complete heritability, brain mass was found to be under diversifying selection with posterior probabilities of $0.0$ for both males and females.
If we assumed that heritability ($\Heritability$) of body mass was uniformly distributed between 20\% and 40\%~\parencite{hu_bringing_2022}, posterior probabilities of diversifying selection became $0.635$ for males and $0.324$ for females.
Mammalian brain mass was found to be under diversifying selection with posterior probabilities of $0.877$ for males and $0.972$ for females when complete heritability was assumed.
Assuming a uniform distribution between 20\% and 40\% for heritability led to posterior probabilities of diversifying selection of $1.0$ for both males and females.

We also analyzed a similar dataset for body mass focusing this time only at Primates (Table~\ref{table:empirical}).
For primates body mass, we found posterior probabilities of diversifying selection of $1.0$ for males and $0.914$ for females when assuming a uniform distribution for the heritability of body mass between 20\% and 40\%.
Assuming complete heritability of body mass did not change the posterior probability for males, but increased the one for female to $1.0$.
Evidence for diversifying selection on body mass was therefore more pronounced in Primates than in mammals.
However, the genetic markers used to normalize trait variance with nucleotide divergence were not necessarily neutral, which could create spurious false positives by artificially inflating $\EstNI$ (Table~\ref{table:assumptions} and methods).

\begin{table*}[t!]
    \centering
    \begin{adjustbox}{width = 1.0\textwidth}
        \begin{tabular}{||l|l||c|c||c|c||}
            \hline
            Broken assumption                                       & Consequences                                       & $\EstRateWhithin$   & $\EstRateBetween$   & Test $\NI > 1$ & Test $\NI < 1$ \\ \hline \hline
            Trait encoded by few loci                        & Between-species trait variation is underestimated & --              & Underestimated & Conservative & Invalid  \\ \hline
            Sexual dimorphism                                & Within-species trait variation is overestimated   & Overestimated & -- & Conservative & Invalid  \\ \hline
            Inbreeding                                       & Nucleotide diversity ($\pi$) is underestimated    & Overestimated  & --              & Conservative & Invalid  \\ \hline
            Markers for polymorphism are negatively selected & Nucleotide diversity ($\pi$) is underestimated  & Overestimated & -- & Conservative & Invalid  \\ \hline
            Markers for polymorphism are positively selected & Nucleotide diversity ($\pi$) is underestimated  & Overestimated & -- & Conservative & Invalid  \\ \hline
            Markers for divergence are positively selected   & Nucleotide divergence ($d$) is overestimated & -- & Underestimated & Conservative & Invalid  \\ \hline
            Markers for polymorphism under balanced selection & Nucleotide diversity ($\pi$) is overestimated  & Underestimated & -- & Invalid & Conservative  \\ \hline
            Markers for divergence are negatively selected   & Nucleotide divergence ($d$) is underestimated & -- & Overestimated & Invalid & Conservative  \\ \hline
            Multiple nucleotide substitutions at the same locus & Nucleotide divergence ($d$) is underestimated & -- & Overestimated & Invalid & Conservative  \\\hline \hline
        \end{tabular}
    \end{adjustbox}
    \caption{Assumptions breaks and their consequences on the estimation of within-species variation ($\EstRateWhithin$), between-species variation ($\EstRateBetween$), and on the neutrality index $\NI = \EstRateBetween/\EstRateWhithin$.
    The last two columns indicate whether the test for diversifying selection ($\NI > 1$) and for stabilizing selection $\NI < 1$ are conservative or invalid due to violated assumptions.
    }
    \label{table:assumptions}
\end{table*}

\section{Discussion}\label{sec:discussion}

% Summary of the method and results
In this study, we proposed a neutrality index for a quantitative trait that can be used within a statistical framework to test for selection.
Our neutrality index for a trait, $\NI$, is calculated as the ratio of the normalized within- to between-species variation and it allowed the identification of the evolutionary regime of a quantitative trait.
At the phylogenetic scale, trait variation between species was normalized by sequence divergence obtained from a neutral set of markers.
Similarly, trait variation within species was normalized by sequence polymorphism obtained also from a neutral set of markers.
Our estimate of $\EstNI$ could be tested for deviation from the value of $1.0$ expected under the null hypothesis of neutrality.
Technically, the neutrality index can be estimated either as a maximum likelihood point estimate, or as a mean posterior estimate from a Bayesian implementation (see section~S\ref{sec:implementation}).
The latter also enabled the estimation of the posterior credible interval to test for departure from a neutrally evolving trait (e.g. $ \proba [ \EstNI > 1 ]$).
We tested our statistical procedure against simulated data and showed that our test was able to correctly detect simulations under diversifying selection (test of $\EstNI > 1$) or under stabilizing selection (test of $\EstNI < 1$).
However, our test detected a spurious signal of stabilizing selection ($\EstNI < 1$) when we simulated the evolution of a neutral trait.
We thus argue that our method should be used to detect diversifying selection, but that it had low accuracy to detect stabilizing selection due to false positives.

% How is our test related to other tests of selection of traits
Our results showed that our method significantly improved over currently available methods to detect selection acting on a trait at the phylogenetic scale.
Current methods relying on evolution of the mean trait value between species also tend to statistically prefer a model of stabilizing selection over a Brownian process when the trait is neutral~\parencite{silvestro_measurement_2015, cooper_cautionary_2016, price_detecting_2022}.
Our approach could in theory be applied to detect stabilizing selection at the phylogenetic scale, but we showed that it did not have the statistical power to identify those cases.
In contrast, we showed that our method was able to identify correctly cases of diversifying selection, which is a clear improvement over current methods that model only mean trait value.
Indeed, under diversifying selection, mean trait value will not deviate from a Brownian process, and thus cannot be distinguished from neutral evolution~\parencite{hansen_translating_1996, harmon_phylogenetic_2018}.
For example, testing the selective regime in the expression level of the majority of genes led to the selection of a Brownian process as the prefered model and the interpretation that the expression was evolving neutrally~\parencite{catalan_drift_2019}.
Our diversity index has the advantage to discriminate the alternative model of diversifying selection from the neutral case by comparing within- and between-species variation correctly normalized to remove confounding factors.
Our approach is not the first one to normalize between-species variation to detect selection, but this was done by using within-species variations~\parencite{rohlfs_modeling_2014, rohlfs_phylogenetic_2015} and not estimates of neutral molecular divergence as done in our study.
These studies have further compared their statistic across a pool of traits, which allowed them to identify outlier traits putatively under diversifying selection but without testing for selection on a single trait at a time~\parencite{rohlfs_phylogenetic_2015, gillard_comparative_2021}.
Instead, our procedure can be applied to a single trait, estimating the neutrality index and giving a statistical test for departures from the null model of neutral evolution for a single test.
Our diversity index opens new avenues to revisit these studies and better test for the selective regime affecting the quantitative traits, assuming we have access to genomic datasets to estimate nucleotide divergence and polymorphism.

% Why is our test different
As such, the main novelty of our study was to use the nucleotide divergence and polymorphism to normalize trait variation between and within species.
In the context of within species variation, \QstFst\ tests have been developed to compare trait and sequence across several populations to test for selection~\parencite{martin_multivariate_2008, leinonen_qst_2013}.
Our neutrality index also used the genetic sequences from which nucleotide divergence and polymorphism are estimated.
Although the sequences should be neutrally evolving, they do not have to be necessarily linked to the quantitative trait under study.
Nucleotide variation allows normalizing for diversity driven by confounding factors such as population sizes ($\Ne$), mutation rates ($\MutationRatePheno$ and $\MutationRateNuc$) and generation time~\parencite{hansen_translating_1996, harmon_phylogenetic_2018}.
Thus our test avoids the estimation of the parameters, which are complex to correctly infer, and it also bypasses the estimation of divergence time, which was necessary in previous approaches~\parencite{walsh_evolution_2018}.
But importantly, by normalizing with sequence variation, we also showed using simulated data that our test was not sensitive to the assumption that $\Ne$, $\MutationRateNuc$ and generation time were constant across the phylogenetic tree, an unmet assumption empirically~\parencite{bergeron_evolution_2023, wilder_contribution_2023}.
In fact, the normalization by nucleotide divergence and polymorphism automatically absorbed long-term and short-term changes in $\Ne$, $\MutationRatePheno$, $\MutationRateNuc$ and generation time, which canceled out in the ratio of trait variation $\EstNI$.

% How is our test related to other tests of selection on sequences
Even though our test was developed for a quantitative trait, analogies with other tests of selection developed for molecular sequences also provided insight into its behavior.
First, we acknowledge that our test took inspiration from the \textcite{mcdonald_adaptative_1991} test devised for protein-coding DNA sequences, where synonymous mutations were used to determine the neutral expectation, and the inflation of divergence was compared to polymorphism within species.
Second, because $\NI$ was compared to 1, our test ultimately bear analogy to the codon-based test of selection, where the ratio of non-synonymous to synonymous substitutions ($\dnds$) is compared to 1~\parencite{goldman_codonbased_1994, muse_likelihood_1994}.
As $\dnds < 1$ is interpreted as purifying selection acting on the protein, $\NI < 1$ is interpreted as stabilizing selection acting on the trait.
Similarly, the interpretation of adaptation for $\dnds > 1$ is analogous to diversifying selection for $\NI > 1$.
With this analogy in mind, we could leverage the vast literature discussing and interpreting the results of these tests and their pitfalls~\parencite{nielsen_molecular_2005, anisimova_investigating_2009, jensen_importance_2019}.
First, not rejecting the neutral null model of $\NI = 1$ did not necessarily imply that the trait was effectively neutral, since diversifying and stabilizing selection could compensate each other resulting in $\NI = 1$, analogously to $\dnds=1$ under a mix of adaptation and purifying selection~\parencite{nielsen_molecular_2005}.
Second, empirical evidence for $\NI < 1$ did not rule out diversifying selection, but rather that this diversifying selection was not strong enough to overcome the stabilizing selection, similarly to strong purifying selection resulting $\dnds < 1$ even though those genes and sites are under adaptation~\parencite{latrille_genes_2023}.
By explicitly modeling stabilizing selection as a moving optimum, it would theoretically be possible to tease apart the effect of diversifying and stabilizing selection in the context of quantitative traits to obtain a statistically more powerful test.

% Pitfalls of the method
In the context of detecting diversifying selection on a trait, we argue that the main drawback of our method is that the additive genetic variance of the trait is required instead of the phenotypic variance.
If phenotypic variance was used instead of additive genetic variance to estimate $\EstNI$, meaning that we assumed complete heritability, the neutrality index $\EstNI$ was ultimately underestimated.
Similarly, using broad-sense heritability instead of narrow-sense heritability would result in underestimated $\EstNI$.
In such context, the test of stabilizing selection ($\EstNI < 1]$) would be statistically invalid.
However, the test of diversifying selection ($\EstNI > 1$) was underpowered although not invalided, meaning that absence of evidence would not be evidence of absence.
As an example, even though we assumed complete heritability for brain mass, we uncovered diversifying selection in mammals since $\EstNI > 1$.

% What can and should be improved
The development of our neutrality index was also based on several assumptions that could be relaxed in future studies.
First, we cannot predict the behavior of our test in the context of population structures, gene flow and introgression.
These factors should be thoroughly investigated using simulations.
Second, loci were assumed to contribute additively to the phenotype.
Although the effects of dominance and epistasis is typically weak compared to the additive effects on the quantitative traits, their influence should be assessed~\parencite{hill_data_2008, crow_epistasis_2010}.
Third, the genetic architecture of the trait was assumed to be constant across the phylogenetic tree, whereas it might actually be variable among individuals and species~\parencite{tung_genetic_2015, huber_conservatism_2015}.
Such an assumption can theoretically be relaxed and changes in genetic architecture along the phylogenetic tree could jointly be estimated~\parencite{arnold_understanding_2008, hohenlohe_mipod_2008, kostikova_bridging_2016, gaboriau_multiplatform_2020}.
Finally, from a statistical perspective, our Bayesian estimation could integrate uncertainty from the estimation of genetic variation, using sequences as input instead of estimated values of nucleotide diversity and divergence.

% What can our test bring to the field
From an empirical point of view, our method required integrating genomic and trait variation, which could reduce the possible datasets to be used.
However, such datasets will become more and more accessible and we showed the applicability of our method by applying it to the illustrative example of mammals brain and body mass.
Because our test was also based on several assumptions that might not hold on empirical data, we also provided a table containing the main assumptions and their consequences on the neutrality index and the test that can be performed (Table~\ref{table:assumptions}).
For example, at the primate scale, the evidence for $\EstNI > 1$ does not necessarily imply that the brain mass was evolving under diversifying selection since the markers used for nucleotide divergences were not neutral, which can lead to a spurious $\EstNI > 1$.
In conclusion, our study provided a statistical framework to test for diversifying selection acting on a quantitative trait while integrating the trove of genomic data available both within and between species, and we believe that our new approach is a promising tool to investigate the evolution of quantitative traits.


%TC:ignore
\printbibliography

\newpage

\part*{Supplementary materials}
\renewcommand{\thetable}{S\arabic{table}}
\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}
\setcounter{table}{0}
\setcounter{section}{0}

\renewcommand{\baselinestretch}{1.0}\normalsize
\tableofcontents
\renewcommand{\baselinestretch}{1.5}\normalsize

\newpage
\section{Genetic architecture of the trait}\label{sec:simulator}

\subsection{Genotype-phenotype map}\label{subsec:genotype-phenotype-map}

\begin{itemize}
    \item $\NbrLoci$ is the number of loci encoding the trait.
    \item $a_l \sim \mathcal{N}(0,a^2)$ is the effect of a mutation on the trait at locus $l \in \{1, \hdots, \NbrLoci\}$.
    \item $\Ne$ is the effective number of individuals.
    \item $g_{i,l} \in \{0, 1, 2\}$ is the genotypic value at locus $l$ for individual $i \in \{1, \hdots, \Ne\}$.
    \item $G_i = \sum_{l=1}^{\NbrLoci} a_l \times g_{i,l}$ is the genotypic value for individual $i$.
    \item $\xi_i \sim \mathcal{N}(0, \VarEnv)$ is the effect of environment on the trait for individual i.
    \item $\Trait_i = G_i + \xi_i$ is the phenotype for individual $i$.
\end{itemize}

\begin{center}
    \captionof{figure}{summary of trait's genetic architecture.}
    \includegraphics[width=0.6\textwidth, page=1] {figureS1}
    \label{fig:simulator-summary}
\end{center}

within-species, the mean ($\bar{G}$) and variance ($\VarGenetic$) of the genotype are:
\begin{equation}
    \bar{G} = \frac{1}{\Ne}\sum_{i=1}^{\Ne} G_i  \text{\quad and \quad} \VarGenetic = \frac{1}{\Ne}\sum_{i=1}^{\Ne}\left(G_i - \bar{G} \right)^2\label{eq:simu-genotype}
\end{equation}
The theoretical additive genetic variance ($\VarGenetic$) is a function of the number of loci ($\NbrLoci$) and the effect of a mutation ($a$) as:
\begin{equation}
    \VarGenetic = 4 \Ne \Multiply \MutationRatePheno \Multiply \NbrLoci \Multiply a^2 \label{eq:simu-var-genetic}
\end{equation}

The mean ($\bar{\Trait}$) and variance ($\VarPhenotype$) of the phenotype are:
\begin{equation}
    \bar{\Trait} = \frac{1}{\Ne}\sum_{i=1}^{\Ne} \Trait_i \text{\quad and \quad} \VarPhenotype = \frac{1}{\Ne}\sum_{i=1}^{\Ne}\left(\Trait_i - \bar{\Trait} \right)^2 \label{eq:simu-between}
\end{equation}

Heritability ($\Heritability$) is defined as:
\begin{equation}
    \Heritability = \frac{\VarGenetic}{\VarPhenotype} = \frac{\VarGenetic}{\VarGenetic + \VarEnv}\label{eq:simu-heritability}
\end{equation}
Altogether, effective population size ($\Ne$), the number of loci ($\NbrLoci$) and the effect of a mutation ($a$), we can compute the variance of the environment ($\VarEnv$) that is required to reach a given heritability ($\Heritability$) as:
\begin{equation}
    \VarEnv = \VarGenetic \Multiply \left( \frac{1}{\Heritability} - 1 \right) = 4 \Ne \Multiply \MutationRatePheno \Multiply \NbrLoci \Multiply a^2 \Multiply \left( \frac{1}{\Heritability} - 1 \right) \label{eq:simu-var-env}
\end{equation}

\newpage
\section{Bayesian estimate}\label{sec:bayesian-estimate}

\subsection{Multivariate Brownian process}\label{subsec:multivariate-brownian-process}
Here we generalize to $\Ntrait$ traits evolving along the phylogeny and are correlated between them.
Their variation along the phylogeny is modeled as a $\Ntrait$-dimensional Brownian process $\Brownian$ ($1 \times \Ntrait$) starting at the root and branching along the tree topology.
The rate of change of the Brownian process is determined by the positive semi-definite and symmetric covariance matrix between traits $\CovarianceMatrix$ ($\Ntrait \times \Ntrait$).
Along branch $j$ with length $d_{j}$, the Brownian process start at the ancestral node $\mathcal{A}(j)$ with value $\Brownian(\mathcal{A}(j))$, and ends at node $\mathcal{R}(j)$  with value $\Brownian(\mathcal{R}(j))$.
The independent contrast $\contrast_{j}$ defined as change in trait along the branch normalized by $\sqrt {d_{j}}$ is a multivariate Gaussian:
\begin{equation}
    \label{eq:DistribBrownian}
    \contrast_{j} = \frac{\Brownian (\mathcal{R}(j)) - \Brownian (\mathcal{A}(j)) }{\sqrt {d_{j}}} \sim \mathcal{N}\left(\VecZero, \CovarianceMatrix \right).
\end{equation}

\subsection{Sampling the covariance matrix}\label{subsec:sampling-the-covariance-matrix}
From the independent contrast at each branch of the tree ($\contrast_{j}$), we can define the $\Ntrait \times \Ntrait$ scatter matrix, $\Scattermatrix$, as:
\begin{equation}
    \Scattermatrix = \sum\limits_{j=1}^{\Nbranch} \contrast_{j} \MultiplyMatrix \left[\contrast_{j}\right]\tr\label{eq:bayes-scatter},
\end{equation}
where $\Nbranch$ is the number of branches in the tree and $\NbrTaxa$ the number of taxa.

The {prior} on the covariance matrix is an inverse Wishart distribution, with $\Ntrait + 1$ degrees of freedom:
\begin{equation}
    \label{eq:Distribcovariance}
    \CovarianceMatrix \sim \text{Wishart}^{-1} (\Identitymatrix, \Ntrait + 1).
\end{equation}

By Bayes theorem, the {posterior} on $\CovarianceMatrix$, conditional on a particular realization of $\Brownian$ (and thus of $\contrast$) is an invert Wishart distribution, of parameter $\Identitymatrix + \Scattermatrix$ and with $\WishartPostDf$ degrees of freedom.
\begin{equation}
    \CovarianceMatrix \sim \text{Wishart}^{-1}\left( \Identitymatrix + \Scattermatrix, \WishartPostDf\right)\label{eq:bayes-posterior}
\end{equation}
This invert Wishart distribution can be obtained by sampling $\WishartPostDf$ independent and identically distributed multivariate normal random variables $\Multivariate_{k}$ defined by
\begin{equation}
    \Multivariate_{k} \sim \mathcal{N} \left( \VecZero, \left[ \Identitymatrix + \Scattermatrix\right]^{-1} \right).\label{eq:bayes-multivariate}
\end{equation}
And from these multivariate samples, $\CovarianceMatrix$ is Gibbs sampled as:
\begin{equation}
    \CovarianceMatrix = \left( \sum\limits_{k=1}^{\WishartPostDf} \Multivariate_{k} \MultiplyMatrix  \left[\Multivariate_{k} \right] \tr \right)\inv \label{eq:bayes-gibbs}
\end{equation}

\newpage
\section{Bayesian and Maximum-likelihood implementation}\label{sec:implementation}

Implementation is included within the \textit{BayesCode} software, available at \censor{\url{https://github.com/ThibaultLatrille/bayescode}}.

\subsection{Data formatting}\label{subsec:data-formatting}

Running the analysis on your dataset and compute posterior probabilities requires three files:
\begin{enumerate}
    \item A phylogenetic tree in newick format, with branch lengths in number of substitutions per site (neutral markers).
    \item A file containing the mean trait values for each species.
    \item A file containing the variation within-species for each trait and the genetic variation within-species (neutral markers).
\end{enumerate}

\subsubsection{Phylogenetic tree}\label{subsubsec:phylogenetic-tree}

The phylogenetic tree must be in newick format, with branch lengths in substitutions per site (neutral markers).

\subsubsection{Mean trait for each species}\label{subsubsec:mean-trait-for-each-species}

The file containing mean trait values for each species must be in a tab-delimited file with the following format:
\begin{center}
    \begin{adjustbox}{width = 0.35\textwidth}
        \begin{tabular}{|l|c|c|}
            \hline
            TaxonName            & Body\_mass & Brain\_mass \\
            \hline
            Panthera\_tigris     & 12.26      & 5.676       \\
            Pithecia\_pithecia   & 7.256      & 3.436       \\
            Colobus\_angolensis  & 9.176      & 4.284       \\
            Saimiri\_boliviensis & 6.845      & 3.279       \\
            $\vdots$             & $\vdots$   & $\vdots$    \\
            \hline
        \end{tabular}\label{tab:trait-mean}
    \end{adjustbox}
\end{center}

The columns are:
\begin{itemize}
    \item \emph{TaxonName}: the name of the taxon matching the name in the alignment and the tree.
    \item As many columns as traits, without spaces or special characters in the trait.
    \item The values can be \texttt{NaN} to indicate that the trait is not available for that taxon.
\end{itemize}

\newpage
\subsubsection{Trait variation for each species}\label{subsubsec:trait-variation-for-each-species}

The file containing trait variation for each species must be in a tab-delimited file with the following format:
\begin{center}
    \begin{adjustbox}{width = 1.0\textwidth}
        \begin{tabular}{|l|c|c|c|c|c|c|}
            \hline
            TaxonName            & Nucleotide\_diversity & Body\_mass\_variance & Body\_mass\_heritability & Brain\_mass\_variance & Brain\_mass\_heritability \\
            \hline
            Pithecia\_pithecia   & 0.0016                & 0.22871              & 0.2                      & 0.00737               & 0.2                       \\
            Colobus\_angolensis  & 0.0017                & 0.00393              & 0.2                      & 0.00416               & 0.2                       \\
            Saimiri\_boliviensis & 0.0013                & 0.00022              & 0.2                      & 0.00045               & 0.2                       \\
            Pygathrix\_nemaeus   & 0.0016                & 0.00347              & 0.2                      & 0.00097               & 0.2                       \\
            $\vdots$             & $\vdots$              & $\vdots$             & $\vdots$                 & $\vdots$              & $\vdots$                  \\
            \hline
        \end{tabular}
        \label{tab:trait-variance}
    \end{adjustbox}
\end{center}

\begin{itemize}
    \item \emph{TaxonName}: the name of the taxon matching the name in the alignment and the tree.
    \item \emph{Nucleotide\_diversity}: the nucleotide diversity within-species (neutral markers), cannot be \texttt{NaN}.
    \item As many columns as traits, without spaces or special characters in the trait.
    \item \emph{TraitName\_variance}: the phenotypic variance of the trait within-species, can be \texttt{NaN} to indicate that the trait variance is not available for that taxon.
    \item \emph{TraitName\_heritability} (optional): the heritability of the trait within-species, between 0 and 1, cannot be \texttt{NaN}.
    \item The columns with the suffix \texttt{\_variance} and \texttt{\_heritability} are repeated for each trait.
    \item \emph{TraitName\_heritability\_lower} (optional): the lower bound of the heritability of the trait within-species, between 0 and 1, cannot be \texttt{NaN}.
    \item \emph{TraitName\_heritability\_upper} (optional): the upper bound of the heritability of the trait within-species, between 0 and 1, cannot be \texttt{NaN}.
    \item If the columns with the suffix \texttt{\_heritability\_lower} and \texttt{\_heritability\_upper} are present, the heritability is randomly drawn from a uniform distribution between the lower and upper bounds.
    \item If the columns with the suffix \texttt{\_heritability} is present, it is taken as is.
    \item If the additive genetic variance (instead of phenotypic variance) is available for a trait, the heritability can be omitted and will automatically be set to 1.0.
\end{itemize}

\newpage
\subsection{Bayesian estimation}\label{subsec:running-nodetraitsand-readnodetraits}

The executable \texttt{nodetraits} from \textit{BayesCode} is used to run the Bayesian estimation of the model, and the executable \texttt{readnodetraits} is used to read the results.

Assuming that the file \texttt{data/body\_size/mammals.male.tsv} contains the mean trait values for each species, the file \texttt{data/body\_size/mammals.male.var\_trait.tsv} contains the variation within-species for each trait and the genetic variation within-species (neutral markers), and the file \texttt{data/body\_size/mammals.male.tree} contains the phylogenetic tree, the following commands are used to run the model and read the results.

\subsubsection{Running the model}\label{subsubsec:running-nodetraits}
\texttt{nodetraits} is run with the following command:
\begin{lstlisting}[language = sh,label={lst:nodetraits-run}]
nodetraits  --until 2000
            --tree data/body_size/mammals.male.tree
            --traitsfile data/body_size/mammals.male.tsv
            run_mammals_male
\end{lstlisting}

\subsubsection{Reading the results}\label{subsubsec:reading-the-results}
Once the model has run, the chain \texttt{run\_mammals\_male} is used to compute the posterior distribution of the ratio of between-species variation over within-species variation with \texttt{readnodetraits}:
\begin{lstlisting}[language = sh,label={lst:readnodetraits-rho}]
readnodetraits --burnin 1000
               --var_within data/body_size/mammals.male.var_trait.tsv
               --output results_mammals_male.tsv
               run_mammals_male
\end{lstlisting}
The file \texttt{data\_empirical/chain\_name.ratio.tsv} then contains the posterior mean of the ratio of between-species variation over within-species variation, the 95\% and 99\% credible interval, and the posterior probability that the ratio is greater than 1.

\subsection{Maximum likelihood estimation}\label{subsec:maximum-likelihood-estimation}

To obtain the ratio (without the posterior credible interval and probability) using maximum likelihood computation, the following python script can be used:
\begin{lstlisting}[language = sh, label={lst:neutrality_index}]
python3 utils/neutrality_index.py --tree data/body_size/mammals.male.tree
                                  --traitsfile data/body_size/mammals.male.tsv
                                  --var_within data/body_size/mammals.male.var_trait.tsv
                                  --output results_ML_mammals_male.tsv
\end{lstlisting}
\end{document}
%TC:endignore