%! BibTeX Compiler = biber
%TC:ignore
\documentclass{article}

\usepackage{xcolor, colortbl}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
linkcolor=DARKBLUELINK,
anchorcolor=DARKBLUELINK,
citecolor=DARKBLUELINK,
filecolor=DARKBLUELINK,
menucolor=DARKBLUELINK,
urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage{biorxiv}
\usepackage[backend=biber,eprint=false,isbn=false,url=false,intitle=true,style=nature,date=year]{biblatex}
\addbibresource{references.bib}

\usepackage{url}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{lmodern}
\usepackage{xfrac, nicefrac}
\usepackage{bm}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{graphicx}
\usepackage{bbold}
\usepackage{pdfpages}
\pdfinclusioncopyfonts=1
\usepackage{lineno}
\renewcommand{\baselinestretch}{1.25}
\renewcommand{\arraystretch}{1.6}

\input{notations.tex}

\title{Testing for neutrality of a trait from within and between species variations}

\author{
\large
\textbf{T. {Latrille}$^{1}$, T. {Gaboriau}$^{1}$, N. {Salamin}$^{1}$}\\
\normalsize
$^{1}$Université de Lausanne, Lausanne, Switzerland\\
\texttt{\href{mailto:thibault.latrille@ens-lyon.org}{thibault.latrille@ens-lyon.org}} \\
}

\begin{document}

\maketitle

% Abstract (≤ 250 words)
%TC:endignore
\begin{abstract}
    To determine whether a trait is neutrally evolving or under selection, methods have been developed both at the population and at the phylogenetic scale.
    At the population scale, methods do not integrate the changes over long evolutionary time.
    Conversely, at the phylogenetic scale, current methods usually discards the variance in traits between individuals, leaving out valuable information.
    The main goal of this project is to combine within-species and between-species variation of quantitative traits to determine the evolutionary regime of a trait.
    We define a neutrality index for a quantitative trait, leveraging both data at the population and phylogenetic scale.
    Moreover, this index leverages sequence divergence and polymorphism to normalize the variation of trait between and within species.
    Our neutrality index equals to one for a neutral trait, suggesting that traits for which this relationship is not verified are putatively under a selective regime.
    If our neutrality index is lower than one for traits putatively under stabilizing selection, and conversely greater than one for trait putatively under diversifying selection.
    Finally, we show that our test is not sensitive to the assumption that population sizes and mutation rates are constant across the phylogeny, and automatically adjust for it.
\end{abstract}

\keywords{Quantitative genetics \and Phylogenetics \and Population-genetics \and Selection }

% Research Article (7500 words)
\section{Introduction}\label{sec:introduction}

Determining whether a particular trait is neutrally evolving or under a particular regime of selection has been a long-standing goal in evolutionary biology.
Fundamentally, distinguishing neutral evolution from selection requires to determine which mode of selection is supported from the observed variation of traits and sequences.
This variation of traits can be observed at different scales, across different development stages at the individual level, across different individuals and populations at the species level, and finally across different species at the phylogenetic level.
All these scales requires different assumptions and methodology, and the endeavor to determine the mode of selection for a given trait has thus incorporated theories, methods and developments across different fields of evolutionary biology such as quantitative-genetics, population-genetics, phylogenetics and comparative genomics\cite{lynch_genetics_1998, walsh_evolution_2018}.
At the population level across individuals, Genome Wide Association Studies (GWAS) in humans have revealed that traits are mostly polygenic (many loci associated to a given trait), pleiotropic (many traits associated to a given loci), additive and mainly under stabilizing selection\cite{simons_population_2018, sella_thinking_2019}.
At the species level, by contrasting variation within and between different populations, studies have shown that stabilizing selection is largely dominant in the evolution of expression of genes\cite{whitehead_neutral_2006, gilad_natural_2006, gilad_expression_2006}.
However, dominant mode of selection acting on gene expression is still controversial and neutral evolution is still debated\cite{signor_evolution_2018, price_detecting_2022}.
For a given population, by contrasting to the divergence to a close species, studies have confirms that changes in gene expression accumulate linearly differences with time from divergence\cite{khaitovich_neutral_2004}.
However, studies including variation across individuals and population typically include species divergence to a close species in a pairwise comparison, and do not integrate the changes across the whole phylogenetic tree.
% This relationship had been proposed to describe the evolutionary regime of a trait for a pair of species\cite{lande_genetic_1980, turelli_heritable_1984}.
% However, it has not yet been generalized for any number of species related by their phylogenetic tree.
% However, this divergence in the level of expression stabilizes over a long distance: the difference between human and chicken (having diverged 300 million years) is similar to that between human and opossum (160 million years).
% This observation suggests that the conservation of essential functions of organs could define a limit for the divergence of transcriptomes.

% Phylogenetics (mean trait evolution)
Contrarily, by accounting for the underlying relationships between species, the mode of selection for a quantitative trait can also be tested at the phylogenetic scale\cite{felsenstein_phylogenies_1985}.
Indeed, if the trait is neutrally evolving, the change in main trait value along a given branch of the tree should be Normally distributed around the ancestral state (at the beginning of branch), with a variance proportional to the length of the branch\cite{hansen_translating_1996}.
As a result, the mean trait value along the whole phylogeny can be modeled as a Brownian process branching at every node of the tree, allowing theoretically to test for neutrality\cite{hansen_translating_1996, harmon_phylogenetic_2018}.
Alternatively, a deviation from the Brownian process, typically an Ornstein-Uhlenbeck process is interpreted as a signature of stabilizing selection\cite{catalan_drift_2019}.
However, studies have shown that comparative approaches are subject to different biases\cite{harmon_phylogenetic_2018}.
First, a better fit for a Brownian process at the phylogenetic scale is not necessarily a proof of the neutral model.
Indeed, a trait under selection for which the optimal trait value is evolving as a Brownian process will not deviate from a Brownian process, and thus be wrongly classified as neutral\cite{hansen_translating_1996}.
Secondly and contrarily, even under a neutral trait, the Ornstein-Uhlenbeck process might sometimes be statistically preferred over a Brownian process due to sampling artifacts\cite{silvestro_measurement_2015, cooper_cautionary_2016, price_detecting_2022}.
Indeed, in such framework, the variation within species is discarded, where the mean trait value is taken either as the trait for a single individual in the species or as the average trait across several individuals.
Altogether, the classical comparative framework allows tests for neutrality at the phylogenetic scale but discards the variance in traits between individuals, leaving out valuable information.

% At the frontier Phylogenetics/Population-genetics
At the frontier between phylogenetics and population genetics, comparative methods at the phylogenetic scale have acknowledged the importance to model within species variation on top of changes in mean trait value\cite{fay_evaluating_2008, kostikova_bridging_2016, gaboriau_multiplatform_2020}.
However, within species variation is used to describe technical errors, or the ratio of between to within variation is estimated for many traits, and is finally compared to the average over all traits to seek deviation from this average\cite{rohlfs_modeling_2014, rohlfs_phylogenetic_2015}.
Determining the evolutionary regime of a quantitative trait through articulation of the variance between and within populations is the goal is this study, while setting threshold for neutral, stabilizing and diversifying selection.
In this study, we propose a neutrality index for a quantitative trait articulating trait variation at both the phylogenetic and population scale.
Importantly, our neutrality index also leverages sequence divergence and polymorphism in order to normalize trait variation at both scales.

\section{Materials and Methods}\label{sec:materials-and-methods}
\subsection{Neutrality index for a quantitative trait}\label{subsec:neutrality-index-for-a-quantitative-trait}
\subsubsection{Mutation-drift equilibrium for a trait}

For a given trait, its genetic architecture is defined by the number of loci encoding the trait ($\NbrLoci$) and the random effect of a mutation on the trait ($a$).
New mutations are thus generating variance for such a trait.
We define the expected effect of mutations on the variance of a trait as $\RateMut$, given as:
\begin{gather}
    \RateMut \defEqual \NbrLoci \Multiply \E \left[ a^2 \right]. \label{eq:rate-mut}
\end{gather}

At the individual level, the mutational variance ($\VarMutation$) is the rate at which new mutations contributes to the trait variance per generation.
As shown in \textcite{lande_quantitative_1979, lande_sexual_1980}, $\VarMutation$ is a function of $\RateMut$ and the mutation rate per loci per generation ($\MutationRate$), as:
\begin{gather}
    \VarMutation = 2 \MutationRate \Multiply \RateMut \label{eq:var-mutation}.
\end{gather}

At the population level, mutations supply new variants in the population, while genetic random drift depletes standing variation.
As shown in \textcite{lynch_mutation_1998}, if the trait is neutral, at equilibrium between mutation and drift, the genetic variance in the population ($\VarGenetic$) is a function of the mutational variance ($\VarMutation$) and the effective number of individual in the population ($\Ne$) as:
\begin{align}
    \VarGenetic & =  2 \Ne \Multiply \VarMutation, \\
    & = 4 \Ne \Multiply \MutationRate \Multiply \RateMut \text{ from eq.~\ref{eq:var-mutation}}\label{eq:var-genetic}.
\end{align}

Moreover, for any neutral genomic region of interest, the genetic diversity, called $\pi$, is measured as the number of mutation segregating in the population divided by the size of the region.
Any of the segregating mutation will eventually reach fixation or extinction due to genetic drift, such that the amount of segregating mutations is also a balance between mutations and drift.
As shown in \textcite{tajima_statistical_1989}, $\pi$ is thus a function of the mutation rate per loci per generation ($\MutationRate$) and the effective population size ($\Ne$), given as:
\begin{gather}
    \pi = 4 \Ne \Multiply \MutationRate \label{eq:genetic-diversity}.
\end{gather}

Altogether, at the population level, we define define $\RatePop$ as the ratio of genetic variance of the trait ($\VarGenetic$) over genetic diversity of any neutral genomic region of interest ($\pi$), which identify with $\RateMut$:
\begin{align}
    \RatePop & \defEqual \frac{\VarGenetic }{\pi}, \\
    & = \frac{4 \Ne \Multiply \MutationRate \Multiply \RateMut}{4 \Ne \Multiply \MutationRate} \text{ from eq.~\ref{eq:var-mutation} and~\ref{eq:genetic-diversity}}, \\
    & = \RateMut. \label{eq:rate-pop}
\end{align}

Additionally, the genetic variance is equal to the observed phenotypic variance ($\VarPhenotype$) multiplied by heredity ($h^2$), giving also $\RatePop$ as:
\begin{align}
    \RatePop & = \frac{h^2 \Multiply \VarPhenotype }{\pi }. \label{eq:rate-pheno-pop}
\end{align}

\subsubsection{Phylogenetic evolution of a trait}

Along a lineage, we denote $\Trait$ as the mean value of the trait over the individuals in the population.
If the trait is neutral and encoded by many loci, $\Trait$ evolves as Brownian process\cite{hansen_translating_1996}.
After the population has evolved for $\Time$ generations, the variance of $\Trait$, called $\VarPhy$, is given by \textcite{hansen_translating_1996} as:
\begin{align}
    \VarPhy & = \frac{\VarGenetic}{\Ne} \Multiply \Time \\
    & = 4 \Time \Multiply \MutationRate \Multiply \RateMut \label{eq:covar},\text{ from eq.~\ref{eq:var-genetic}},
\end{align}

Moreover, for any neutral genomic region of interest, out of all neutral mutations that originated in the populations, some will we eventually reach fixation in the population due to genetic drift, resulting in a substitution.
For a neutral mutation, its probability of fixation ($\pfix$) is $1/2\Ne$, as shown in \textcite{kimura_probability_1962}.
As reviewed in \textcite{mccandlish_modeling_2014}, we can thus derive the substitution rate per generation, denoted $\SubRate$ as the number of mutations per generation ($2\Ne \Multiply \MutationRate$) multiplied by the probability of fixation for each of these newly arisen mutations $\pfix$, giving:
\begin{align}
    \SubRate & = 2 \Ne \Multiply \MutationRate \Multiply \pfix, \\
    & = 2 \Ne  \Multiply \MutationRate  \Multiply \dfrac{1}{2\Ne}, \\
    & = \MutationRate. \label{eq:substitution-rate}
\end{align}
That is, if mutations are neutral, the rate with which a new neutral allele is fixed in the population in a genomic region of interest equals the rate with which new mutations arise per generation for the same genomic region\cite{kimura_evolutionary_1968}.

Thus, after having evolved for $\Time$ generations, the fraction of the genomic region of interest that generated a substitution, called $d$, will be given as the number of generations ($\Time$) multiplied by the substitution rate per generation ($\SubRate$), given as:
\begin{align}
    d & = \Time \Multiply \SubRate \\
    & = \Time \Multiply \MutationRate \text{ from eq.~\ref{eq:substitution-rate}}. \label{eq:distance}
\end{align}
Altogether, the variance in the mean trait value ($\VarPhy$) normalized by the genetic distance of any neutral genomic region of interest ($d$) also identify with $\RateMut$:
\begin{align}
    \RatePhy & \defEqual \frac{\VarPhy}{4 d},\\
    & = \frac{4 \Time \Multiply \MutationRate \Multiply \RateMut}{4 \Time \Multiply \MutationRate}\text{ from eq.~\ref{eq:covar} and~\ref{eq:distance}}, \\
    & = \RateMut \label{eq:rate-phy}.
\end{align}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=\textwidth, page=1] {artworks/fig-summary}
    \caption{
        In red, at the phylogenetic scale, the variance of mean trait value ($\VarPhy$) normalised by genetic distance ($d$) is defined as $\RatePhy$.
        In blue, at the population scale, trait variance within species ($\VarGenetic$) normalised by genetic diversity ($\pi$) is defined as $\RatePop$.
        Under neutral evolution $\RatePhy$ should equal $\RatePop$.
        Importantly, the sequence from which $\pi$ and $d$ are estimated should be neutrally evolving, but they are not necessarily linked to the quantitative trait under study, they allow to normalize for diversity driven by mutation rate and population size.
    }
    \label{fig:methods}
\end{figure*}

\subsubsection{Neutrality index}

 The variability between individuals or between species can be obtained for both quantitative trait and genomic sequences.
 At the population level, the variability of the trait between individuals allows to compute $\VarGenetic$, and the variability and of any genomic region neutrally evolving allow to compute $\pi$.
Combined as a ratio, we obtain $\RatePop$ at the population level, which is a proxy for $\RateMut$, namely the expected effect of a mutation on the variance of a trait.
At the phylogenetic level, the variability of the mean trait value between species allows to compute $\VarPhy$, and the variability and of any genomic region neutrally evolving allow to compute $d$.
Similarly, combined as a ratio, we obtain $\RatePhy$ which is another proxy for $\RateMut$.
Altogether, we thus have:
\begin{align}
    \RateMut & = \RatePop = \RatePhy \text{ from eq.~\ref{eq:rate-pop} and~\ref{eq:rate-phy}} , \\
    \Rightarrow \frac{\RatePhy}{\RatePop} & = \frac{\VarPhy \Multiply \pi}{ \VarGenetic \Multiply 4 d }  = 1.
\end{align}
Altogether, $\RatePhy / \RatePop$ is our neutrality index, which should identify to $1$ for a trait neutrally evolving.
Importantly, $\RatePhy$ and $\RatePop$ can both be estimated using quantitative trait and genomic sequences within and between populations, while neither the mutation rate $\MutationRate$ nor the effective population size $\Ne$ need to be estimated.

\subsection{Estimate}

\subsubsection{Maximum likelihood estimate}
For each population of interest with inter-individual data available, $\RatePop$ can be estimated as the ratio of genetic variance of a trait over genetic diversity of neutrally evolving sequences.
Averaged across all populations we obtain the mean as:
\begin{gather}
    \EstRatePop = \left< \frac{\VarGenetic}{\pi} \right> = \left< \frac{h^2 \Multiply \VarPhenotype }{\pi } \right> \label{eq:estimated-rate-pop}
\end{gather}

At the phylogenetic scale, for $\NbrSpecies$ taxa in the tree, $\Cr$ is the $\NbrSpecies \times \NbrSpecies$ contrast matrix computed from the branch lengths ($d$) and the topology of the phylogenetic tree.
The contrast matrix is symmetric, along the diagonal are the total distances ($d_{\Spi}$) from the root of the tree to each taxon ($\Spi$).
The off-diagonal elements $C_{\Spi,\Spj}$ are the total branch lengths shared by particular pairs of taxa $\Spi$ and $\Spj$, computed from the distance between the pair ($d_{\Spi,\Spj}$) as:
\begin{gather}
    C_{\Spi,\Spj} = d_{\mathcal{A}(\Spi, \Spj)} = \frac{d_{\Spi} + d_{\Spj} - d_{\Spi,\Spj}}{2}
\end{gather}
Then, the root state for the character, called $\RootTrait$, can be estimated from the $\NbrSpecies \times 1$ vector of mean trait values for tip species in the tree, called $\VecTrait$, as a maximum likelihood estimate\cite{omeara_testing_2006}, given as:
\begin{gather}
    \RootTrait = \left( \VecOne\tr \Cr\inv \VecOne \right)\inv \left( \VecOne\tr \Cr\inv \VecTrait \right), \label{eq:estimated-root-trait}
\end{gather}
where $\VecOne$ is an $\NbrSpecies \times 1$ column vector of ones.

Finally, the normalized mutational variance at the phylogenetic scale ($\EstRatePhy$) is estimated\cite{omeara_testing_2006} as:
\begin{gather}
    \EstRatePhy = \frac{1}{4}\frac{\left( \VecTrait -  \RootTrait \VecOne \right)\tr \Cr\inv \left( \VecTrait -  \RootTrait  \VecOne  \right)}{\NbrSpecies - 1}. \label{eq:estimated-rate-phy}
\end{gather}
Importantly, the sequence from which $\pi$ and $d$ are estimated should be neutrally evolving, but they are not necessarily linked to the quantitative trait under study, they allow to normalize for diversity driven by mutation rate and population size.

\subsection{Simulation}\label{subsec:simulations}
The evolutionary dynamics was formalized as a Wright-Fisher model with mutation, selection and drift.
The population is assumed to be panmictic, with {effective population size} $\Ne$ and with non-overlapping generations.
We simulate explicitly each generation along the phylogeny under a Wright-Fisher population, consisting of three steps: mutation, selection and {genetic drift} of currently segregating alleles.
Mutations are drawn based on a user-defined nucleotide matrix, where our simulations used a symmetric time-reversible mutation matrix.
Drift is induced by the multinomial resampling of the currently segregating alleles.
Our simulator use a geometric Brownian multivariate process to model the changes in the mutation rate per generation, the generation time and $\Ne$ along the lineages.
Moreover, the instant value of log-$\Ne$ can also be modelled as a sum of a geometric Brownian process and an Ornstein-Uhlenbeck process.
The geometric Brownian motion accounts for long-term fluctuations, while the Ornstein-Uhlenbeck introduces short-term fluctuations.
In our simulations, the tree is composed of 77 species (see supplementary), the tree root is $150$ million years old, the initial mutation rate is $\smash{10^{-8}}$ per site per generation and the initial generation time is $10$ years.
The simulator starts from an initial sequence at equilibrium, at each node, the process is split, and finally stopped at the leaves of the tree.

We introduced selection as a one dimensional Fisher's geometric landscape~\cite{tenaillon_utility_2014,blanquart_epistasis_2016}.
The phenotype is a vector in a multidimensional space, where the number of dimensions is often termed complexity.
From a {phenotype}, the fitness is a monotonously decreasing function of the {phenotype} distance to $0$.
The exact functional phenotype-fitness map depends on $2$ external parameters controlling for strength ($\alpha$) and epistasis ($\beta$).
If the phenotype-fitness map is explicit, the genotype-phenotype map is more pervasive.
Mutations are seen has displacement of the {phenotype} in the multidimensional space.
Beneficial mutations are moving the {phenotype} closer to $0$, whereas deleterious mutations are moving the {phenotype} further away.
In such original context, the distribution of mutational effects is not dependent on the current genotype, but this can be relaxed using a genotype-phenotype map.

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=\textwidth, page=1] {artworks/fig-simulator}
    \caption{
        Wright-Fisher simulations with mutation, selection and drift.
    }
    \label{fig:simulator}
\end{figure*}

\section{Results}

To determine whether a trait is neutrally evolving or under selection, we reviewed the theoretical equations of trait evolution at the population and phylogenetic scale.
The underlying mutational input for a neutral trait is formally related to both within and between species variation of the trait.
More specifically, for a heritable trait under a neutral regime, the genetic variance of the trait ($\VarGenetic$) normalized by genetic diversity of any neutral genomic region ($\pi$) is a proxy for the mutational variance ($\RatePop$).
At the phylogenetic level, along a specific lineage, the mean value for the trait evolves as a Brownian process, and the variance in this mean value after a time $\Time$ is called $\VarPhy$.
This variance can also be normalized by the genetic distance of any neutral genomic region ($d$), allowing us to obtain another proxy for the mutational variance ($\RatePhy$).
Altogether, we define the neutrality index as $\RatePhy/\RatePop$, which equals to $1$ for a neutral trait, suggesting that traits for which this relationship is not verified are putatively under a selective regime.
Under a depleted variation between species, meaning $\RatePhy < \RatePop$, the trait is putatively under stabilizing, forcing the mean value to be homogeneous between species.
In contrast, under an inflated variation between species, meaning $\RatePhy > \RatePop$, the trait is putatively under diversifying selection, or a moving optimum such that the mean value is heterogeneous in the different species.

\begin{table*}[tb]
    \centering
    \begin{adjustbox}{width = 0.75\textwidth}
        \begin{tabular}{|c||c|c|}
            \hline
            & Coding sequences                                  & Quantitative traits                                               \\ \hline \hline
            Adaptation or moving optimum       & $\frac{\dn \Multiply \ps}{\ds \Multiply \pn} > 1$ & $ \frac{\VarPhy \Multiply \pi}{ \VarGenetic \Multiply 4 d } > 1 $ \\ \hline
            Neutral regime of evolution        & $\frac{\dn \Multiply \ps}{\ds \Multiply \pn} = 1$ & $ \frac{\VarPhy \Multiply \pi}{ \VarGenetic \Multiply 4 d } = 1 $ \\ \hline
            Purifying or directional selection & $\frac{\dn \Multiply \ps}{\ds \Multiply \pn} < 1$ & $\frac{\VarPhy \Multiply \pi}{ \VarGenetic \Multiply 4 d } < 1 $ \\ \hline
        \end{tabular}
    \end{adjustbox}
    \caption{
        The relationship between trait variation between and within species can be used as method to detect whether traits are evolving under a neutral model of evolution.
        Similarly, in molecular evolution, non-synonymous divergence ($\dn$) and polymorphism $\pn$ are normalized by synonymous divergence ($\ds$) and polymorphism ($\ps$) to produce a neutrality index\cite{mcdonald_adaptative_1991, fay_evaluating_2008}.
    }
    \label{table:unfolded-MK}
\end{table*}

Importantly, the genetic sequences from which $\pi$ and $d$ are estimated should be neutrally evolving, but they are not necessarily linked to the quantitative trait under study, they allow to normalize for diversity driven by mutation rate and population size.
Analogously to the \textcite{mcdonald_adaptative_1991} test devised for protein-coding DNA sequences, a category of neutral mutations (synonymous) is used to determine the neutral expectation, and the inflation of divergence is compared to polymorphism within species.
Altogether, our neutrality index for a quantitative trait leverages the data for any number of species, and takes advantage of the signal over the whole phylogeny, while at the same time taking into account phylogenetic inertia and addressing the non-independence between species (see fig.~\ref{fig:methods}).
This statistic can be obtained as a maximum likelihood estimate, from eq.~\ref{eq:estimated-rate-pop} and~\ref{eq:estimated-rate-phy}.

\subsection{Results against simulations}

The inference framework was first tested on independently simulated datasets.
With the aim of applying the inference method to empirical datasets, the simulation parameters were chosen so as to match an empirically relevant empirical regime.
Thus, the tree topology and the branch lengths were chosen based on a tree estimated on the mammalian dataset further considered below.

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=\textwidth, page=1] {artworks/constant_pop_size_phy_pop.hist}
    \includegraphics[width=\textwidth, page=1] {artworks/fluctuating_pop_size_phy_pop.hist}
    \caption{
        $\RatePhy$ estimated at the phylogenetic scale as a function of $\RatePop$ estimated at the population scale, for $30.000$ genes simulated under different evolutionary regimes.
        $\NIx < 1$ for traits simulated under selection (stabilizing selection in yellow).
        $\NIx = 1$ for traits simulated under a neutral evolution (in blue).
        $\NIx > 1$ for genes simulated under a moving optimum (diversifying selection in red).
        Effective population size ($\Ne$) and mutation rate $\MutationRate$ are either constant (top panel), or fluctuating as a Brownian process along the phylogeny (panel B).
    }
    \label{fig:constant_pop_size_phy_pop}
\end{figure*}

\section{Discussion}\label{sec:discussion}

In this study, we proposed a neutrality index for a quantitative trait.
At the phylogenetic scale, trait variation between species is normalized by sequence divergence obtained from a neutral set of markers.
Similarly, at the population scale, trait variation within species is normalized by sequence polymorphism obtained from a neutral set of markers.
Our neutrality index for a trait is the ratio of between species variation to within species variation.
Such neutrality index can be compared to 1 to seek a statistical deviation from a neutral trait.
In other words, our statistical test does to seeks to find traits with extremal values in comparison to a pool of traits.
Instead, our procedure can be applied to a single trait, estimating the neutrality index and giving a statistical test for departure from the null model of neutral evolution.
Technically, the neutrality index is estimated either as a maximum likelihood point estimate, or as mean posterior from a Bayesian implementation also giving access to the posterior credible interval allowing to test for a departure from a neutrally evolving trait.
Importantly, our neutrality index leverages sequence divergence and polymorphism, as well as the heritability of traits.
We argue that the main novelty of this study is being able to use sequence divergence and polymorphism to normalize the variation of trait between and within species.
By doing so, our test is not sensitive to the assumption that population sizes ($\Ne$) and mutation rates ($\MutationRate$) are constant across the phylogeny, not requiring to estimate divergence time\cite{litsios_effects_2012}.
Instead, the normalization by sequence divergence and polymorphism automatically absorbs these changes\cite{seo_estimating_2004}.
Contrarily, we argue that the main drawback of our method is that the heritability ($h^2$) of a trait is required if one has access only to the phenotypic variance ($\VarPhenotype$) and not the genetic variance ($\VarGenetic$), which is empirically hard to measure for any trait.

At the phylogenetic scale, the change in mean trait value across different species is typically used to test for neutrality for a quantitative trait.
These models are seeking deviation from a Brownian process\cite{catalan_drift_2019}, or from an Ornstein–Uhlenbeck process accounting for within-species variation\cite{rohlfs_phylogenetic_2015}.
For example, the analysis on the transcriptome across different individuals and species concluded that the expression level of 81\% of genes are under a neutral regime and 16\% under directional selection\cite{catalan_drift_2019}.
However, the alternative model of a varying directional or stabilizing selection was not implemented, and thus can not be ruled out by this analysis.
With our statistical test, we can set a threshold for neutrality and thus are able to make the difference between a trait evolving neutrally as a Brownian process or a trait under selection where the optimal is moving as a Brownian.
Our diversity index can thus allow to revisit these studies, assuming we have access to genomic datasets to estimate divergence ($d$) and polymorphism ($\pi$), as well as estimate for heritabilities ($h^2$).

% Assumption that could be relaxed: constant genetic architecture and multidimensional generalization
More theoretically, our development is based on several assumption that could be relaxed.
First, $\RateMut$ is assumed to be constant across the phylogeny, while it is determined by the underlying genetic architecture of a trait, which can be variable among individuals and species.
Having a constant $\RateMut$ across the phylogeny is an assumption that can theoretically by relaxed by adding a time component\cite{arnold_understanding_2008, hohenlohe_mipod_2008}, where $\RateMut(t)$ could jointly be estimated along the phylogeny\cite{kostikova_bridging_2016, gaboriau_multiplatform_2020}.
Moreover, even though the neutrality index can be applied for has many traits as available, each trait is so far assumed to be independent.
Assuming covariation between traits is also an extension that can be implemented within the multidimensional Brownian framework\cite{huelsenbeck_detecting_2003,  lartillot_phylogenetic_2011,lartillot_joint_2012}.
Altogether, we argue that our neutrality index is a promising statistical procedure allowing to determine the evolutionary regime of a quantitative trait, through articulation of the variance between and within populations in a unified framework.

%TC:ignore
\section*{Acknowledgements}
\label{sec:acknowledgment}
\textbf{Funding:}
Université de Lausanne.
\textbf{Author contributions:}
Original idea: T.L.\ and N.S.;
Model conception: T.L.\ and N.S.;
Code: T.L.;
Data analyses: T.L.;
Interpretation: T.L., T.G\ and N.S.;
First draft: T.L.;
Editing and revisions: T.L., T.G\ and N.S.;
Project management and funding: N.S\@.
\textbf{Competing interests:}
The authors declare no conflicts of interest.
\textbf{Data and materials availability:}
Snakemake pipeline, analysis scripts and documentation are available at \href{https://github.com/ThibaultLatrille/MicMac}{github.com/ThibaultLatrille/MicMac}.

\printbibliography

\end{document}
%TC:endignore